// @algolia/ui-library/public/esm/constants/external-routes.js
export const ACADEMY = 'https://academy.algolia.com/';
export const BLOG = '/blog/';
export const COMMUNITY = 'https://community.algolia.com';
export const DASHBOARD = '/dashboard';
export const DOC = '/doc';
export const DOC_CHANGELOG = `${DOC}/changelog/`;
export const SEARCH_AND_DISCOVERY = `${DOC}/guides/solutions/ecommerce/ui-kits/`;
export const DISCOURSE = 'https://discourse.algolia.com';
export const DOCSEARCH = 'https://docsearch.algolia.com';
export const FAQ = `${DOC}/faq/`;
export const GETTING_STARTED = `${DOC}/guides/getting-started/how-algolia-works/`;
export const GITHUB = 'https://github.com/algolia';
export const GRADER = 'https://grader.algolia.com';
export const NEWSLETTER = 'https://go.algolia.com/subscription_newsletter/';
export const PARTNERS = 'https://partners.algolia.com';
export const RESOURCES = 'https://resources.algolia.com';
export const RESOURCES_CASE_STUDIES = `${RESOURCES}/customer-stories`;
export const SHOW_AND_TELL = 'https://discourse.algolia.com/c/show-tell';
export const SIGN_IN = '/users/sign_in';
export const SIGN_OUT = '/users/sign_out';
export const SIGN_UP = '/users/sign_up';
export const STATUS = 'https://status.algolia.com';
export const STORIES = 'https://stories.algolia.com';
export const SUPPORT = '/support/';
export const TALKSEARCH = 'https://community.algolia.com/talksearch/';
export const TECH_BLOG = '/tech-blog/';
export const YOUTUBE = 'https://www.youtube.com/user/Algolia';
export const ZENDESK_GET_STARTED = '/getstarted#/zendesk';
// Temporary constants, remove when not used anymore
export const WEBINAR_SUMMER_19 = '/webinar/summer-19/';
export const WEBINAR_WINTER_20 = 'https://go.algolia.com/webinar-winter-20/';
