// @algolia/ui-library/public/esm/constants/routes.js
export const ZENDESK = '/integrations/zendesk/';
export const ACCEPTABLE_USE = '/policies/acceptable-use/';
export const COOKIES = '/policies/cookies/';
export const INFRASTRUCTURE_AND_SUBPROCESSORS = '/policies/infrastructure-and-sub-processors/';
export const LEGAL = '/policies/legal/';
export const POLICIES_OLD = '/policies/old/';
export const PRIVACY = '/policies/privacy/';
export const SLA = '/policies/sla/';
export const POLICIES_SUPPORT = '/policies/support/';
export const TERMS = '/policies/terms/';
const SEARCH_AND_DISCOVERY = '/products/search-and-discovery/';
const DISTRIBUTED_SECURE = '/distributed-secure/';
export const AB_TESTING = `${SEARCH_AND_DISCOVERY}ab-testing/`;
export const ANALYTICS = `${SEARCH_AND_DISCOVERY}analytics/`;
export const ANSWERS = `${SEARCH_AND_DISCOVERY}answers/`;
export const ANSWERS_FOR_CUSTOMER_SUPPORT = `${SEARCH_AND_DISCOVERY}answers-for-customer-support/`;
export const ANSWERS_FOR_PUBLISHERS = `${SEARCH_AND_DISCOVERY}answers-for-publishers/`;
export const ENTERPRISE = `${SEARCH_AND_DISCOVERY}enterprise/`;
export const HEADLESS_COMMERCE = `${SEARCH_AND_DISCOVERY}headless-commerce/`;
export const HOSTED_SEARCH_API = `${SEARCH_AND_DISCOVERY}hosted-search-api/`;
export const MOBILE_SEARCH = `${SEARCH_AND_DISCOVERY}mobile-search/`;
export const PERSONALIZATION = `${SEARCH_AND_DISCOVERY}personalization/`;
export const PERSONALIZED_QUERY_SUGGESTIONS = `${SEARCH_AND_DISCOVERY}personalized-query-suggestions/`;
export const SEARCH_RELEVANCE_RULES = `${SEARCH_AND_DISCOVERY}search-relevance-rules/`;
export const SITE_SEARCH = `${SEARCH_AND_DISCOVERY}site-search/`;
export const UI_COMPONENT_LIBRARIES = `${SEARCH_AND_DISCOVERY}ui-component-libraries/`;
export const VOICE_SEARCH = `${SEARCH_AND_DISCOVERY}voice-search/`;
export const VISUAL_MERCHANDISING_CURATION = `${SEARCH_AND_DISCOVERY}visual-merchandising-curation/`;
export const PRODUCTS_SEARCH = `${SEARCH_AND_DISCOVERY}hosted-search-api/`;
export const AI_SEARCH = `/products/ai-search/`;
export const DYNAMIC_SYNONYM_SUGGESTIONS = `${AI_SEARCH}dynamic-synonym-suggestions/`;
export const DYNAMIC_RERANKING = `${AI_SEARCH}dynamic-reranking/`;
export const AI_STUDIO = `/products/ai-studio/`;
export const CRAWLER = `/products/crawler/`;
export const RECOMMENDATIONS = `/products/recommendations/`;
export const WHATS_NEW = `/products/whats-new/`;
export const SUMMER_19 = `/products/whats-new/summer19/`;
const SEARCH_SOLUTIONS = '/search-solutions/';
export const AZURE = `${SEARCH_SOLUTIONS}algolia-on-azure/`;
export const B2B_ECOMMERCE = `${SEARCH_SOLUTIONS}b2b-ecommerce/`;
export const ECOMMERCE = `${SEARCH_SOLUTIONS}ecommerce/`;
export const GEO_SEARCH = `${SEARCH_SOLUTIONS}geo-search/`;
export const MAGENTO = `${SEARCH_SOLUTIONS}magento/`;
export const MARKETPLACE_SEARCH = `${SEARCH_SOLUTIONS}marketplaces/`;
export const MEDIA = `${SEARCH_SOLUTIONS}media/`;
export const NETLIFY = `${SEARCH_SOLUTIONS}netlify/`;
export const SAAS = `${SEARCH_SOLUTIONS}saas/`;
export const SECURITY = `${DISTRIBUTED_SECURE}security-compliance/`;
export const INFRASTRUCTURE = `${DISTRIBUTED_SECURE}global-infrastructure/`;
export const SECURITY_MEASURES = `${SECURITY}/measures/`;
export const SALESFORCE_COMMERCE_CLOUD = `${SEARCH_SOLUTIONS}salesforce-commerce-cloud/`;
export const SHOPIFY = `${SEARCH_SOLUTIONS}shopify/`;
export const CUSTOMERS = '/resources/customers/';
export const ABOUT = '/about/';
export const ALGOLIA_VS_ELASTICSEARCH = '/algolia-vs-elasticsearch/';
export const CAREERS = '/careers/';
export const COMPARE_ALGOLIA_VS_ELASTICSEARCH = '/compare-algolia-vs-elasticsearch/';
export const CONTACT_US = '/contactus/';
export const CONTACT_SUPPORT = '/support/?contact=';
export const COVID_19 = '/covid-19/';
export const DEMO_REQUEST = '/demorequest/';
export const DEVELOPERS = '/developers/';
export const DEVELOPERS_GET_STARTED = `${DEVELOPERS}get-started`;
export const CONTENTFUL_SEARCH = `${DEVELOPERS}contentful-search-algolia`;
export const ERROR = '404';
export const EVENTS = '/events/';
export const FOR_OPEN_SOURCE = '/for-open-source/';
export const HOME = '/';
export const INTEGRATIONS = `${DEVELOPERS}#integrations`;
export const PRESS = '/press/';
export const PRICING = '/pricing/';
export const PRODUCTS = '/products/';
export const REDEEM = '/redeem/';
export const SEARCH = '/search/';
export const SEARCH_INSPIRATION_LIBRARY = '/search-inspiration-library/';
export const SOCIAL_IMPACT = '/social-impact/';
export const STATE_OF_SEARCH = '/state-of-site-search-and-discovery/';
export const TECH_BLOG = '/tech-blog/';
export const WHY = '/why/';
export const SIGN_UP_EMBED = '/embed/sign-up/';
export const ABOUT_YOU_EMBED = '/embed/about-you/';
export const BUSINESS_INFORMATION_EMBED = '/embed/business-information/';
export const CHECK_EMAIL_EMBED = '/embed/check-email/';
