// @algolia/ui-library/public/components/Carousel/Carousel.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const breakpoints = __importStar(require("../../constants/breakpoints"));
exports.default = {
    slide: react_1.css `
    transition: transform 0.2s ease-out, z-index 0.2s ease-out;
    transform-origin: center;
  `,
    getLeftSlide: (isWide) => react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-87%' : '-116%'});
      opacity: 0.3;
    }
    @media (min-width: ${breakpoints.LG}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-87%' : '-149%'});
    }
  `,
    middleSlide: react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      transform: translateX(-50%);
      z-index: 1;
    }
  `,
    getRightSlide: (isWide) => react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-38%' : '-17%'});
      opacity: 0.3;
    }
    @media (min-width: ${breakpoints.LG}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-38%' : '16%'});
    }
  `,
    buttonLeft: react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      right: calc(100% + 20px);
      transform: translate(0, -50%);
      transition: transform 0.2s ease-in-out;
      &:hover {
        transform: translate(-2px, -50%);
      }
    }
  `,
    buttonRight: react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      left: calc(100% + 20px);
      transform: translate(0, -50%);
      transition: transform 0.2s ease-in-out;
      &:hover {
        transform: translate(2px, -50%);
      }
    }
  `,
    dotsContainer: react_1.css `
    bottom: -48px;
  `,
};
