// @algolia/ui-library/public/esm/components/CustomersLogos/CustomersLogos.data.js
export default {
    '6play': {
        alt: '6play',
        src: '6play.svg',
        height: 40,
    },
    '8tracks': {
        alt: '8Tracks',
        src: '8tracks.svg',
        height: 40,
    },
    abc: {
        alt: 'ABC',
        src: 'abc.svg',
        height: 30,
    },
    absolunet: {
        alt: 'Absolunet',
        src: 'absolunet.png',
        height: 100,
    },
    accenture: {
        alt: 'Accenture',
        src: 'accenture.svg',
        height: 40,
    },
    adobe: {
        alt: 'Adobe',
        src: 'adobe.svg',
        height: 70,
    },
    aftership: {
        alt: 'Aftership',
        src: 'aftership.svg',
        height: 40,
    },
    akqa: {
        alt: 'Akqa',
        src: 'akqa.svg',
        height: 40,
    },
    alltrails: {
        alt: 'Alltrails',
        src: 'alltrails.svg',
        height: 40,
    },
    'america-test-kitchen': {
        alt: 'America’s Test Kitchen',
        src: 'america-test-kitchen.png',
        height: 50,
    },
    amplitude: {
        alt: 'Amplitude',
        src: 'amplitude.svg',
        height: 40,
    },
    angellist: {
        alt: 'AngelList',
        src: 'angellist.svg',
        height: 60,
    },
    aptdeco: {
        alt: 'AptDeco',
        src: 'aptdeco.svg',
        height: 30,
    },
    arcteryx: {
        alt: "Arc'teryx",
        src: 'arcteryx.svg',
        height: 30,
    },
    astound: {
        alt: 'Astound',
        src: 'astound.svg',
        height: 40,
    },
    atlassian: {
        alt: 'Atlassian',
        src: 'atlassian.svg',
        height: 25,
    },
    balance: {
        alt: 'Balance',
        src: 'balance.svg',
        height: 30,
    },
    balinea: {
        alt: 'Balinea',
        src: 'balinea.svg',
        height: 30,
    },
    bd: {
        alt: 'BD',
        src: 'bd.png',
        height: 100,
    },
    betterworldbooks: {
        alt: 'Better World Books',
        src: 'betterworldbooks.svg',
        height: 40,
    },
    birchbox: {
        alt: 'Birchbox',
        src: 'birchbox.svg',
        height: 20,
    },
    blablacar: {
        alt: 'BlaBlaCar',
        src: 'blablacar.svg',
        height: 30,
    },
    blissim: {
        alt: 'Blissim ex-Birchbox',
        src: 'blissim.svg',
        height: 50,
    },
    bombas: {
        alt: 'Bombas',
        src: 'bombas.svg',
        height: 53,
    },
    boots: {
        alt: 'Boots',
        src: 'boots.svg',
        height: 80,
    },
    brillio: {
        alt: 'Brillio',
        src: 'brillio.png',
        height: 50,
    },
    bringmeister: {
        alt: 'Bringmeister',
        src: 'bringmeister.png',
        height: 30,
    },
    'bringmeister-classic': {
        alt: 'Bringmeister',
        src: 'bringmeister-classic.svg',
        height: 30,
    },
    'blue-acorn-ici': {
        alt: 'Blue Acorn iCi',
        src: 'blue-acorn-ici.png',
        height: 30,
    },
    cafeyn: {
        alt: 'Cafeyn',
        src: 'cafeyn.svg',
        height: 60,
    },
    callrail: {
        alt: 'CallRail',
        src: 'callrail.png',
        height: 50,
    },
    capgemini: {
        alt: 'Capgemini',
        src: 'capgemini.svg',
        height: 40,
    },
    caudalie: {
        alt: 'Caudalie',
        src: 'caudalie.svg',
        height: 50,
    },
    'cleveland-clinic': {
        alt: 'Cleveland Clinic',
        src: 'cleveland-clinic.svg',
        height: 30,
    },
    cnetric: {
        alt: 'Cnetric',
        src: 'cnetric.svg',
        height: 50,
    },
    concord: {
        alt: 'Concord',
        src: 'concord.svg',
        height: 50,
    },
    couchbase: {
        alt: 'Couchbase',
        src: 'couchbase.svg',
        height: 40,
    },
    coursera: {
        alt: 'Coursera',
        src: 'coursera.svg',
        height: 25,
    },
    'culture-king': {
        alt: 'Culture King',
        src: 'culture-king.svg',
        height: 53,
    },
    curtsy: {
        alt: 'Curtsy',
        src: 'curtsy.png',
        height: 50,
    },
    dailymotion: {
        alt: 'Dailymotion',
        src: 'dailymotion.svg',
        height: 100,
    },
    dashlane: {
        alt: 'Dashlane',
        src: 'dashlane.svg',
        height: 50,
    },
    decathlon: {
        alt: 'Decathlon',
        src: 'decathlon.svg',
        height: 40,
    },
    'decathlon-singapore': {
        alt: 'Decathlon Singapore',
        src: 'decathlon-singapore.png',
        height: 40,
    },
    dentsu: {
        alt: 'Dentsu',
        src: 'dentsu.svg',
        height: 30,
    },
    depop: {
        alt: 'Depop',
        src: 'depop.svg',
        height: 40,
    },
    digitalOcean: {
        alt: 'DigitalOcean',
        src: 'digitalocean.svg',
        height: 40,
    },
    dior: {
        alt: 'Dior',
        src: 'dior.svg',
        height: 40,
    },
    discovery: {
        alt: 'Discovery',
        src: 'discovery.svg',
        height: 40,
    },
    docker: {
        alt: 'Docker',
        src: 'docker.svg',
        height: 40,
    },
    dover: {
        alt: 'Dover',
        src: 'dover.svg',
        height: 40,
    },
    dubsmash: {
        alt: 'Dubsmash',
        src: 'dubsmash.png',
        height: 50,
    },
    dunelm: {
        alt: 'Dunelm',
        src: 'dunelm.svg',
        height: 50,
    },
    evolution7: {
        alt: 'Evolution 7',
        src: 'evolution7.svg',
        height: 30,
    },
    'eyebuy-direct': {
        alt: 'EyeBuy Direct',
        src: 'eyebuy-direct.svg',
        height: 50,
    },
    fanatics: {
        alt: 'Fanatics',
        src: 'fanatics.svg',
        height: 40,
    },
    flexport: {
        alt: 'Flexport',
        src: 'flexport.svg',
        height: 25,
    },
    forrester: {
        alt: 'Forrester',
        src: 'forrester.svg',
        height: 50,
    },
    francetv: {
        alt: 'France TV',
        src: 'francetv.svg',
        height: 30,
    },
    garmin: {
        alt: 'Garmin TV',
        src: 'garmin.svg',
        height: 60,
    },
    geckoboard: {
        alt: 'Geckoboard',
        src: 'geckoboard.svg',
        height: 50,
    },
    genius: {
        alt: 'Genius',
        src: 'genius.svg',
        height: 25,
    },
    goat: {
        alt: 'GOAT',
        src: 'goat.svg',
        height: 25,
    },
    gofundme: {
        alt: 'gofundme',
        src: 'gofundme.svg',
        height: 50,
    },
    'go-sport': {
        alt: 'Go Sport',
        src: 'go-sport.svg',
        height: 50,
    },
    guerin: {
        alt: 'Guérin',
        src: 'guerin.svg',
        height: 40,
    },
    gymshark: {
        alt: 'Gymshark',
        src: 'gymshark.svg',
        height: 30,
    },
    hobbyking: {
        alt: 'HobbyKing',
        src: 'hobbyking.svg',
        height: 40,
    },
    'ibis-world': {
        alt: 'Ibis World',
        src: 'ibis-world.svg',
        height: 60,
    },
    ibmix: {
        alt: 'Ibmix',
        src: 'ibmix.svg',
        height: 30,
    },
    idealist: {
        alt: 'Idealist',
        src: 'idealist.svg',
        height: 30,
    },
    iflix: {
        alt: 'Iflix',
        src: 'iflix.svg',
        height: 80,
    },
    influenster: {
        alt: 'Influenster',
        src: 'influenster.svg',
        height: 40,
    },
    intuit: {
        alt: 'Intuit',
        src: 'intuit.svg',
        height: 40,
    },
    jbhifi: {
        alt: 'JB Hi-Fi',
        src: 'jbhifi.svg',
        height: 25,
    },
    keepsmallstrong: {
        alt: 'Keep Small Strong',
        src: 'keep-small-strong.png',
        height: 20,
    },
    kensium: {
        alt: 'Kensium',
        src: 'kensium.svg',
        height: 40,
    },
    kitsune: {
        alt: 'Maison Kitsuné',
        src: 'kitsune.svg',
        height: 40,
    },
    kooples: {
        alt: 'The Kooples',
        src: 'kooples.svg',
        height: 30,
    },
    lacoste: {
        alt: 'Lacoste',
        src: 'lacoste.svg',
        height: 30,
    },
    lancaster: {
        alt: 'Lancaster',
        src: 'lancaster.svg',
        height: 30,
    },
    legalzoom: {
        alt: 'Legalzoom',
        src: 'legalzoom.svg',
        height: 40,
    },
    lekiosk: {
        alt: 'LeKiosk',
        src: 'lekiosk.png',
        height: 50,
    },
    livearea: {
        alt: 'LiveArea',
        src: 'livearea.png',
        height: 30,
    },
    loccitane: {
        alt: "L'Occitane",
        src: 'loccitane.svg',
        height: 40,
    },
    louboutin: {
        alt: 'Louboutin',
        src: 'louboutin.svg',
        height: 70,
    },
    lvmh: {
        alt: 'LVMH',
        src: 'lvmh.svg',
        height: 30,
    },
    magento: {
        alt: 'Magento',
        src: 'magento-adobe.svg',
        height: 60,
    },
    manomano: {
        alt: 'ManoMano',
        src: 'manomano.png',
        height: 60,
    },
    mckinsey: {
        alt: 'McKinsey',
        src: 'mckinsey.svg',
        height: 50,
    },
    medium: {
        alt: 'Medium',
        src: 'medium.svg',
        height: 30,
    },
    mercadona: {
        alt: 'Mercadona',
        src: 'mercadona.svg',
        height: 30,
    },
    mercari: {
        alt: 'Mercari',
        src: 'mercari.svg',
        height: 60,
    },
    'moda-operandi': {
        alt: 'Moda Operandi',
        src: 'moda-operandi.svg',
        height: 40,
    },
    'national-geographic': {
        alt: 'National Geographic',
        src: 'national-geographic.svg',
        height: 50,
    },
    nbc: {
        alt: 'NBC',
        src: 'nbc.svg',
        height: 53,
    },
    nbcu: {
        alt: 'NBCUniversal',
        src: 'nbcu.svg',
        height: 25,
    },
    nintendo: {
        alt: 'Nintendo',
        src: 'nintendo.svg',
        height: 40,
    },
    npr: {
        alt: 'NPR',
        src: 'npr.svg',
        height: 40,
    },
    'ouest-france': {
        alt: 'Ouest France',
        src: 'ouest-france.svg',
        height: 50,
    },
    overdose: {
        alt: 'Overdose',
        src: 'overdose.png',
        height: 30,
    },
    pentair: {
        alt: 'Pentair',
        src: 'pentair.svg',
        height: 50,
    },
    'place-it': {
        alt: 'Place it',
        src: 'place-it.svg',
        height: 53,
    },
    politico: {
        alt: 'Politico',
        src: 'politico.svg',
        height: 30,
    },
    'publicis-sapient': {
        alt: 'Publicis Sapient',
        src: 'publicis-sapient.png',
        height: 50,
    },
    pubnub: {
        alt: 'Pubnub',
        src: 'pubnub.png',
        height: 30,
    },
    r2i: {
        alt: 'r2i',
        src: 'r2i.png',
        height: 60,
    },
    radley: {
        alt: 'Radley',
        src: 'radley.svg',
        height: 40,
    },
    raja: {
        alt: 'Raja',
        src: 'raja.png',
        height: 30,
    },
    redbox: {
        alt: 'Redbox',
        src: 'redbox.svg',
        height: 30,
    },
    rugs: {
        alt: 'Rugs',
        src: 'rugs.svg',
        height: 53,
    },
    rxsaver: {
        alt: 'RxSaver',
        src: 'rxsaver.svg',
        height: 40,
    },
    salesforce: {
        alt: 'Salesforce',
        src: 'salesforce.svg',
        height: 70,
    },
    'salesforce-commerce-cloud': {
        alt: 'Salesforce Commerce Cloud',
        src: 'salesforce-commerce-cloud.png',
        height: 90,
    },
    'salling-group': {
        alt: 'Salling Group',
        src: 'salling-group.svg',
        height: 40,
    },
    'sante-discount': {
        alt: 'Santé Discount logo',
        src: 'sante-discount.svg',
        height: 30,
    },
    sense: {
        alt: 'Sense',
        src: 'sense.svg',
        height: 25,
    },
    shipstation: {
        alt: 'ShipStation',
        src: 'shipstation.svg',
        height: 30,
    },
    shopify: {
        alt: 'Shopify',
        src: 'shopify.svg',
        height: 50,
    },
    slack: {
        alt: 'Slack',
        src: 'slack.svg',
        height: 40,
    },
    'something-digital': {
        alt: 'Something Digital',
        src: 'something-digital.svg',
        height: 40,
    },
    staples: {
        alt: 'Staples',
        src: 'staples.svg',
        height: 30,
    },
    steunze: {
        alt: 'Steun Ze',
        src: 'steunze.png',
        height: 70,
    },
    strava: {
        alt: 'Strava',
        src: 'strava.svg',
        height: 30,
    },
    streema: {
        alt: 'Streema',
        src: 'streema.png',
        height: 30,
    },
    stripe: {
        alt: 'Stripe',
        src: 'stripe.svg',
        height: 40,
    },
    'teachers-pay-teachers': {
        alt: 'Teachers Pay Teachers',
        src: 'teachers-pay-teachers.svg',
        height: 53,
    },
    teespring: {
        alt: 'Teespring',
        src: 'teespring.svg',
        height: 30,
    },
    'the-real-real': {
        alt: 'TheRealReal',
        src: 'the-real-real.svg',
        height: 40,
    },
    'the-trade-desk': {
        alt: 'The Trade Desk',
        src: 'the-trade-desk.svg',
        height: 30,
    },
    twitch: {
        alt: 'Twitch',
        src: 'twitch.svg',
        height: 40,
    },
    ubisoft: {
        alt: 'Ubisoft',
        src: 'ubisoft.svg',
        height: 50,
    },
    'under-armour': {
        alt: 'Under Armour',
        src: 'under-armour.svg',
        height: 30,
    },
    uniqlo: {
        alt: 'Uniqlo',
        src: 'uniqlo.svg',
        height: 60,
    },
    universal: {
        alt: 'Universal',
        src: 'universal.svg',
        height: 60,
    },
    vaangroup: {
        alt: 'Vaangroup',
        src: 'vaangroup.svg',
        height: 50,
    },
    vaimo: {
        alt: 'Vaimo',
        src: 'vaimo.png',
        height: 50,
    },
    'vestiaire-collective': {
        alt: 'Vestiaire Collective',
        src: 'vestiaire-collective.svg',
        height: 40,
    },
    weightwatchers: {
        alt: 'Weight Watcher',
        src: 'weightwatchers.svg',
        height: 60,
    },
    wework: {
        alt: 'Wework',
        src: 'wework.svg',
        height: 30,
    },
    wisembly: {
        alt: 'Wisembly',
        src: 'wisembly.svg',
        height: 40,
    },
    zaelab: {
        alt: 'Zaelab',
        src: 'zaelab.svg',
        height: 40,
    },
    zendesk: {
        alt: 'Zendesk',
        src: 'zendesk.svg',
        height: 30,
    },
    zilker: {
        alt: 'Zilker',
        src: 'zilker.svg',
        height: 40,
    },
};
