// @algolia/ui-library/public/esm/components/Footer/Banner/Product/index.js
/** @jsxRuntime classic /
/** @jsx jsx */
import { Fragment } from 'react';
import { jsx } from '@emotion/react';
import * as routes from '../../../../constants/routes';
import * as extRoutes from '../../../../constants/external-routes';
import { getDefaultHref, useDemoButtonLink } from '../utils';
import styles from '../Banner.css';
import Heading2 from '../../../Heading2';
import Button from '../../../Button';
import ProductCard from './ProductCard';
const PRODUCT_BANNER_DATA = {
    search: {
        text: 'Enable anyone to build great Search & Discovery',
        ctaBackground: 'search',
        ctaLabel: 'Start building for free >_',
    },
    recommend: {
        text: 'Build Recommend experiences',
        ctaBackground: 'recommend',
        ctaLabel: 'Start building for free >_',
    },
    answers: {
        text: 'Algolia Answers',
        ctaBackground: 'answers',
        ctaLabel: 'Start building for free >_',
    },
};
const Product = ({ type, internalLinkTag, externalRoutes, withUrlPrefix, ctaProps, trackEvent, }) => {
    const buttonLinkProps = useDemoButtonLink({
        internalLinkTag,
        ctaProps,
        trackEvent,
        defaultHref: getDefaultHref(routes.DEMO_REQUEST, withUrlPrefix),
    });
    const { text, ctaBackground, ctaLabel } = PRODUCT_BANNER_DATA[type];
    return (jsx(Fragment, null,
        jsx("div", { css: styles[type], className: "uil-ph-32 uil-pt-70 uil-pos-relative" },
            jsx("div", { className: "uil-maw-1440 uil-m-auto" },
                jsx("div", { className: "uil-mb-48 uil-ai-center uil-d-flex uil-fxd-column sm:uil-d-grid sm:uil-g-2 " },
                    jsx(Heading2, { className: "uil-d-flex uil-jc-between uil-m-auto uil-mb-20 uil-lsp-normal uil-ta-center sm:uil-mb-16 sm:uil-m-0 md:uil-ta-left md:uil-mr-70 md:uil-mb-0", color: "white" },
                        jsx("span", { className: "uil-fsz-24 uil-fw-semibold uil-lh-small sm:uil-fsz-36" }, text)),
                    jsx("div", { className: "uil-d-flex uil-fxd-column uil-ai-center sm:uil-ai-end xl:uil-fxd-row" },
                        jsx(Button, Object.assign({ className: "uil-mb-16 xl:uil-mb-0 xl:uil-mr-16" }, buttonLinkProps), "Get a demo"),
                        jsx(Button, { background: ctaBackground, color: "white", href: extRoutes.SIGN_UP }, ctaLabel))))),
        jsx("div", { css: styles.cardContainer, className: "uil-bgc-grey-800 uil-ph-32 uil-pb-20 md:uil-pb-70" },
            jsx("div", { className: "uil-maw-1440 uil-m-auto" },
                jsx(ProductCard, { internalLinkTag: internalLinkTag, externalRoutes: externalRoutes, withUrlPrefix: withUrlPrefix })))));
};
export default Product;
