// @algolia/fragments.js/lib/modules/typography/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "fontFamily", {
    enumerable: true,
    get: function() {
        return _fontFamily.default;
    }
});
Object.defineProperty(exports, "fontSize", {
    enumerable: true,
    get: function() {
        return _fontSize.default;
    }
});
Object.defineProperty(exports, "fontStyle", {
    enumerable: true,
    get: function() {
        return _fontStyle.default;
    }
});
Object.defineProperty(exports, "fontWeight", {
    enumerable: true,
    get: function() {
        return _fontWeight.default;
    }
});
Object.defineProperty(exports, "letterSpacing", {
    enumerable: true,
    get: function() {
        return _letterSpacing.default;
    }
});
Object.defineProperty(exports, "lineClamp", {
    enumerable: true,
    get: function() {
        return _lineClamp.default;
    }
});
Object.defineProperty(exports, "lineHeight", {
    enumerable: true,
    get: function() {
        return _lineHeight.default;
    }
});
Object.defineProperty(exports, "listStyle", {
    enumerable: true,
    get: function() {
        return _listStyle.default;
    }
});
Object.defineProperty(exports, "textAlign", {
    enumerable: true,
    get: function() {
        return _textAlign.default;
    }
});
Object.defineProperty(exports, "textDecoration", {
    enumerable: true,
    get: function() {
        return _textDecoration.default;
    }
});
Object.defineProperty(exports, "textOverflow", {
    enumerable: true,
    get: function() {
        return _textOverflow.default;
    }
});
Object.defineProperty(exports, "textTransform", {
    enumerable: true,
    get: function() {
        return _textTransform.default;
    }
});
Object.defineProperty(exports, "whiteSpace", {
    enumerable: true,
    get: function() {
        return _whiteSpace.default;
    }
});
Object.defineProperty(exports, "wordBreak", {
    enumerable: true,
    get: function() {
        return _wordBreak.default;
    }
});
var _fontFamily = _interopRequireDefault(require("./font-family"));
var _fontSize = _interopRequireDefault(require("./font-size"));
var _fontStyle = _interopRequireDefault(require("./font-style"));
var _fontWeight = _interopRequireDefault(require("./font-weight"));
var _letterSpacing = _interopRequireDefault(require("./letter-spacing"));
var _lineClamp = _interopRequireDefault(require("./line-clamp"));
var _lineHeight = _interopRequireDefault(require("./line-height"));
var _listStyle = _interopRequireDefault(require("./list-style"));
var _textAlign = _interopRequireDefault(require("./text-align"));
var _textDecoration = _interopRequireDefault(require("./text-decoration"));
var _textOverflow = _interopRequireDefault(require("./text-overflow"));
var _textTransform = _interopRequireDefault(require("./text-transform"));
var _whiteSpace = _interopRequireDefault(require("./white-space"));
var _wordBreak = _interopRequireDefault(require("./word-break"));
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
