// @algolia/ui-library/public/components/UserGeneratedContent/UserGeneratedContent.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const colors = __importStar(require("../../constants/colors"));
exports.default = {
    root: react_1.css `
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      color: ${colors.GREY_900};
    }
    a {
      color: ${colors.NEBULA_500};
      text-decoration: none;
    }
    table {
      text-align: left;
      border-spacing: 0;
      width: 100%;
      border: 1px solid ${colors.GREY_200};
    }
    thead th {
      padding: 8px;
      border-bottom: 1px solid ${colors.GREY_200};
    }
    tbody > *:nth-child(odd) {
      background-color: ${colors.WHITE};
    }
    tbody > *:nth-child(even) {
      background-color: ${colors.GREY_100};
    }
    tbody td {
      font-size: 14px;
      padding: 8px;
    }
    *:first-child {
      margin-top: 0;
    }
    *:last-child {
      margin-bottom: 0;
    }
    ul {
      li p {
        margin: 0;
      }
    }
  `,
};
