// @algolia/fragments.js/lib/modules/box/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "boxSizing", {
    enumerable: true,
    get: function() {
        return _boxSizing.default;
    }
});
Object.defineProperty(exports, "display", {
    enumerable: true,
    get: function() {
        return _display.default;
    }
});
Object.defineProperty(exports, "margin", {
    enumerable: true,
    get: function() {
        return _margin.default;
    }
});
Object.defineProperty(exports, "overflow", {
    enumerable: true,
    get: function() {
        return _overflow.default;
    }
});
Object.defineProperty(exports, "padding", {
    enumerable: true,
    get: function() {
        return _padding.default;
    }
});
Object.defineProperty(exports, "visibility", {
    enumerable: true,
    get: function() {
        return _visibility.default;
    }
});
var _boxSizing = _interopRequireDefault(require("./box-sizing"));
var _display = _interopRequireDefault(require("./display"));
var _margin = _interopRequireDefault(require("./margin"));
var _overflow = _interopRequireDefault(require("./overflow"));
var _padding = _interopRequireDefault(require("./padding"));
var _visibility = _interopRequireDefault(require("./visibility"));
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
