// @algolia/ui-library/public/components/CodeSample/LanguagesList/LanguagesList.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const colors = __importStar(require("../../../constants/colors"));
const styles = {
    languageListContainer: react_1.css `
    background: #090b1e;
    border-top-left-radius: 4px;
    border-bottom-left-radius: 4px;
  `,
    languagesList: react_1.css `
    grid-template-columns: 26px 1fr;
  `,
    getCursorStyle: (color, cursorOffsetY) => react_1.css `
    transform: translateY(${cursorOffsetY}px);
    background: linear-gradient(
      90deg,
      ${color === 'blue' ? colors.NEBULA_500 : colors.GREEN_700} 0%,
      rgb(23, 26, 48) 92.71%
    );
    transition: transform 0.2s ease-in-out;
  `,
};
exports.default = styles;
