// @algolia/ui-library/public/components/Button/Button.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const breakpoints = __importStar(require("../../constants/breakpoints"));
const colors = __importStar(require("../../constants/colors"));
const generateStates = (color, shadow = true) => react_1.css `
    ${shadow &&
    react_1.css `
        box-shadow: 0px 2px 4px rgba(45, 35, 66, 0.4),
          0px 7px 13px -3px rgba(45, 35, 66, 0.3), inset 0px -3px 0px ${color};
      `}
    &:hover {
      box-shadow: 0px 4px 8px rgba(45, 35, 66, 0.4),
        0px 7px 13px -3px rgba(45, 35, 66, 0.3), inset 0px -3px 0px ${color};
      transform: translateY(-2px);
    }
    &:focus {
      box-shadow: inset 0 0 0 1.5px ${color}, 0px 2px 4px rgba(45, 35, 66, 0.4),
        0px 7px 13px -3px rgba(45, 35, 66, 0.3), inset 0px -3px 0px ${color};
    }
    &:active {
      box-shadow: inset 0px 3px 7px ${color};
      transform: translateY(2px);
    }
  `;
exports.default = {
    root: react_1.css `
    height: 42px;
    transition: box-shadow 0.15s ease, transform 0.15s ease;
    will-change: box-shadow, transform;

    @media (min-width: ${breakpoints.LG}) {
      height: 48px;
    }
  `,
    text: react_1.css `
    top: -2px;
  `,
    disabled: react_1.css `
    transform: none !important;
  `,
    blue: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.CYAN_400} 0%,
      ${colors.NEBULA_500} 100%
    );
    box-shadow: 0px 2px 4px rgba(45, 35, 66, 0.4),
      0px 7px 13px -3px rgba(45, 35, 66, 0.3),
      inset 0px -3px 0px rgba(58, 65, 111, 0.5);
    ${generateStates(colors.NEBULA_600, false)}
  `,
    pink: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.PINK_300} 0%,
      ${colors.PINK_600} 100%
    );
    ${generateStates(colors.PINK_700)}
  `,
    white: react_1.css `
    background: ${colors.GREY_050};
    ${generateStates(colors.GREY_200)}
  `,
    orange: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.ORANGE_500} 0%,
      ${colors.ORANGE_700} 100%
    );
    ${generateStates(colors.ORANGE_800)}
  `,
    green: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.GREEN_500} 0%,
      ${colors.GREEN_700} 100%
    );
    ${generateStates(colors.GREEN_800)}
  `,
    purple: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.PINK_200} 0%,
      ${colors.COLOR_ANSWER} 100%
    );
    ${generateStates('rgb(0 0 0 / 20%)')}
  `,
    red: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.ORANGE_400} 0%,
      ${colors.COLOR_RECOMMEND} 100%
    );
    ${generateStates('rgb(0 0 0 / 20%)')}
  `,
    dark: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.GREY_500} 0%,
      ${colors.GREY_700} 100%
    );
    ${generateStates(colors.GREY_900)}
  `,
    answers: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.PINK_200} 0%,
      ${colors.COLOR_ANSWER} 100%
    );
    ${generateStates('rgba(54, 57, 90, 0.2)')}
  `,
    search: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.CYAN_400} 0%,
      ${colors.NEBULA_500} 100%
    );
    ${generateStates('rgba(54, 57, 90, 0.2)')}
  `,
    recommend: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.ORANGE_400} 0%,
      ${colors.COLOR_RECOMMEND} 100%
    );
    ${generateStates('rgba(54, 57, 90, 0.2)')}
  `,
    round: react_1.css `
    width: 42px;
    @media (min-width: ${breakpoints.LG}) {
      width: 48px;
    }
  `,
    shine: react_1.css `
    text-shadow: 0 1px 0 rgba(0, 0, 0, 0.4);
    &::before {
      content: '';
      position: absolute;
      z-index: 1;
      left: 0;
      top: 0;
      width: var(--size);
      height: var(--size);
      background: radial-gradient(
        circle closest-side,
        var(--shineColor),
        transparent
      );
      transform: translate(calc(-50% + var(--x)), calc(-50% + var(--y)));
      transition: height 0.2s ease;
      will-change: width, height, transform;
    }
  `,
    shineIcon: react_1.css `
    filter: drop-shadow(0 1px 0 rgba(0, 0, 0, 0.4));
  `,
};
