// @algolia/ui-library/public/components/Footer/Banner/Product/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
/** @jsxRuntime classic /
/** @jsx jsx */
const react_1 = require("react");
const react_2 = require("@emotion/react");
const routes = __importStar(require("../../../../constants/routes"));
const extRoutes = __importStar(require("../../../../constants/external-routes"));
const utils_1 = require("../utils");
const Banner_css_1 = __importDefault(require("../Banner.css"));
const Heading2_1 = __importDefault(require("../../../Heading2"));
const Button_1 = __importDefault(require("../../../Button"));
const ProductCard_1 = __importDefault(require("./ProductCard"));
const PRODUCT_BANNER_DATA = {
    search: {
        text: 'Enable anyone to build great Search & Discovery',
        ctaBackground: 'search',
        ctaLabel: 'Start building for free >_',
    },
    recommend: {
        text: 'Build Recommend experiences',
        ctaBackground: 'recommend',
        ctaLabel: 'Start building for free >_',
    },
    answers: {
        text: 'Algolia Answers',
        ctaBackground: 'answers',
        ctaLabel: 'Start building for free >_',
    },
};
const Product = ({ type, internalLinkTag, externalRoutes, withUrlPrefix, ctaProps, trackEvent, }) => {
    const buttonLinkProps = utils_1.useDemoButtonLink({
        internalLinkTag,
        ctaProps,
        trackEvent,
        defaultHref: utils_1.getDefaultHref(routes.DEMO_REQUEST, withUrlPrefix),
    });
    const { text, ctaBackground, ctaLabel } = PRODUCT_BANNER_DATA[type];
    return (react_2.jsx(react_1.Fragment, null,
        react_2.jsx("div", { css: Banner_css_1.default[type], className: "uil-ph-32 uil-pt-70 uil-pos-relative" },
            react_2.jsx("div", { className: "uil-maw-1440 uil-m-auto" },
                react_2.jsx("div", { className: "uil-mb-48 uil-ai-center uil-d-flex uil-fxd-column sm:uil-d-grid sm:uil-g-2 " },
                    react_2.jsx(Heading2_1.default, { className: "uil-d-flex uil-jc-between uil-m-auto uil-mb-20 uil-lsp-normal uil-ta-center sm:uil-mb-16 sm:uil-m-0 md:uil-ta-left md:uil-mr-70 md:uil-mb-0", color: "white" },
                        react_2.jsx("span", { className: "uil-fsz-24 uil-fw-semibold uil-lh-small sm:uil-fsz-36" }, text)),
                    react_2.jsx("div", { className: "uil-d-flex uil-fxd-column uil-ai-center sm:uil-ai-end xl:uil-fxd-row" },
                        react_2.jsx(Button_1.default, Object.assign({ className: "uil-mb-16 xl:uil-mb-0 xl:uil-mr-16" }, buttonLinkProps), "Get a demo"),
                        react_2.jsx(Button_1.default, { background: ctaBackground, color: "white", href: extRoutes.SIGN_UP }, ctaLabel))))),
        react_2.jsx("div", { css: Banner_css_1.default.cardContainer, className: "uil-bgc-grey-800 uil-ph-32 uil-pb-20 md:uil-pb-70" },
            react_2.jsx("div", { className: "uil-maw-1440 uil-m-auto" },
                react_2.jsx(ProductCard_1.default, { internalLinkTag: internalLinkTag, externalRoutes: externalRoutes, withUrlPrefix: withUrlPrefix })))));
};
exports.default = Product;
