// @algolia/ui-library/public/esm/layouts/FeaturesRow/FeaturesRow.css.js
import { css } from '@emotion/react';
import * as breakpoints from '../../constants/breakpoints';
import * as colors from '../../constants/colors';
export default {
    root: css `
    transition: transform 0.2s ease-out, box-shadow 0.2s ease-out;
  `,
    light: css `
    background-image: linear-gradient(${colors.WHITE}, ${colors.GREY_100});
  `,
    dark: css `
    background-image: radial-gradient(
      circle at 50% 0,
      ${colors.GREY_700},
      ${colors.GREY_900}
    );
  `,
    fadedCard: css `
    @media (min-width: ${breakpoints.SM}) {
      &::before {
        content: '';
        position: absolute;
        right: -1px;
        top: 0;
        height: 60px;
        background: linear-gradient(to top, rgba(255, 255, 255, 0), white 100%);
        width: 1px;
      }
      &::after {
        content: '';
        position: absolute;
        right: -1px;
        bottom: 0;
        height: 60px;
        background: linear-gradient(
          to bottom,
          rgba(255, 255, 255, 0),
          white 100%
        );
        width: 1px;
      }
    }
  `,
};
