// @algolia/fragments.js/lib/modules/misc/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "appearance", {
    enumerable: true,
    get: function() {
        return _appearance.default;
    }
});
Object.defineProperty(exports, "boxShadow", {
    enumerable: true,
    get: function() {
        return _boxShadow.default;
    }
});
Object.defineProperty(exports, "cursor", {
    enumerable: true,
    get: function() {
        return _cursor.default;
    }
});
Object.defineProperty(exports, "opacity", {
    enumerable: true,
    get: function() {
        return _opacity.default;
    }
});
Object.defineProperty(exports, "pointerEvents", {
    enumerable: true,
    get: function() {
        return _pointerEvents.default;
    }
});
Object.defineProperty(exports, "resize", {
    enumerable: true,
    get: function() {
        return _resize.default;
    }
});
Object.defineProperty(exports, "strokeWidth", {
    enumerable: true,
    get: function() {
        return _strokeWidth.default;
    }
});
Object.defineProperty(exports, "transition", {
    enumerable: true,
    get: function() {
        return _transition.default;
    }
});
Object.defineProperty(exports, "userSelect", {
    enumerable: true,
    get: function() {
        return _userSelect.default;
    }
});
var _appearance = _interopRequireDefault(require("./appearance"));
var _boxShadow = _interopRequireDefault(require("./box-shadow"));
var _cursor = _interopRequireDefault(require("./cursor"));
var _opacity = _interopRequireDefault(require("./opacity"));
var _pointerEvents = _interopRequireDefault(require("./pointer-events"));
var _resize = _interopRequireDefault(require("./resize"));
var _strokeWidth = _interopRequireDefault(require("./strokeWidth"));
var _transition = _interopRequireDefault(require("./transition"));
var _userSelect = _interopRequireDefault(require("./user-select"));
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
