// @algolia/ui-library/public/esm/components/CodeSample/types.js
/* eslint-disable no-shadow */
export var BackLanguage;
(function (BackLanguage) {
    BackLanguage["Ruby"] = "ruby";
    BackLanguage["Rails"] = "rails";
    BackLanguage["Python"] = "python";
    BackLanguage["Django"] = "django";
    BackLanguage["Php"] = "php";
    BackLanguage["Symphony"] = "symphony";
    BackLanguage["Laravel"] = "laravel";
    BackLanguage["JavascriptBack"] = "javascript-back";
    BackLanguage["Java"] = "java";
    BackLanguage["Scala"] = "scala";
    BackLanguage["Go"] = "go";
    BackLanguage["CSharp"] = "c-sharp";
    BackLanguage["Kotlin"] = "kotlin";
    BackLanguage["Swift"] = "swift";
    BackLanguage["AndroidBack"] = "android";
})(BackLanguage || (BackLanguage = {}));
export var FrontLanguage;
(function (FrontLanguage) {
    FrontLanguage["JavascriptFront"] = "javascript-front";
    FrontLanguage["React"] = "react";
    FrontLanguage["AndroidFront"] = "android-front";
    FrontLanguage["Vue"] = "vue";
    FrontLanguage["Angular"] = "angular";
    FrontLanguage["Ios"] = "ios";
})(FrontLanguage || (FrontLanguage = {}));
export const Language = {
    ...BackLanguage,
    ...FrontLanguage,
};
