// @algolia/ui-library/public/esm/components/Footer/Banner/Banner.css.js
import { css } from '@emotion/react';
import * as colors from '../../../constants/colors';
import * as breakpoints from '../../../constants/breakpoints';
export default {
    default: css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.CYAN_400} 0%,
      ${colors.NEBULA_500} 100%
    );
  `,
    search: css `
    padding-bottom: 10rem;
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.CYAN_400} 0%,
      ${colors.NEBULA_500} 100%
    );
  `,
    answers: css `
    padding-bottom: 10rem;
    background: radial-gradient(
      100% 100% at 100% 0%,
      #cb97ff 0%,
      ${colors.COLOR_ANSWER} 100%
    );
  `,
    recommend: css `
    padding-bottom: 10rem;
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.PINK_300} 0%,
      ${colors.COLOR_RECOMMEND} 100%
    );
  `,
    card: css `
    background: linear-gradient(
      180deg,
      ${colors.GREY_600} 0%,
      ${colors.GREY_700} 100%
    );
    box-shadow: inset 0px -2px 4px rgba(0, 0, 0, 0.08);
  `,
    cardContainer: css `
    margin-top: -10rem;

    @media (min-width: ${breakpoints.SM}) {
      margin-top: -11rem;
    }
  `,
};
