// @algolia/ui-library/public/components/CodeSample/types.js
"use strict";
/* eslint-disable no-shadow */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Language = exports.FrontLanguage = exports.BackLanguage = void 0;
var BackLanguage;
(function (BackLanguage) {
    BackLanguage["Ruby"] = "ruby";
    BackLanguage["Rails"] = "rails";
    BackLanguage["Python"] = "python";
    BackLanguage["Django"] = "django";
    BackLanguage["Php"] = "php";
    BackLanguage["Symphony"] = "symphony";
    BackLanguage["Laravel"] = "laravel";
    BackLanguage["JavascriptBack"] = "javascript-back";
    BackLanguage["Java"] = "java";
    BackLanguage["Scala"] = "scala";
    BackLanguage["Go"] = "go";
    BackLanguage["CSharp"] = "c-sharp";
    BackLanguage["Kotlin"] = "kotlin";
    BackLanguage["Swift"] = "swift";
    BackLanguage["AndroidBack"] = "android";
})(BackLanguage = exports.BackLanguage || (exports.BackLanguage = {}));
var FrontLanguage;
(function (FrontLanguage) {
    FrontLanguage["JavascriptFront"] = "javascript-front";
    FrontLanguage["React"] = "react";
    FrontLanguage["AndroidFront"] = "android-front";
    FrontLanguage["Vue"] = "vue";
    FrontLanguage["Angular"] = "angular";
    FrontLanguage["Ios"] = "ios";
})(FrontLanguage = exports.FrontLanguage || (exports.FrontLanguage = {}));
exports.Language = {
    ...BackLanguage,
    ...FrontLanguage,
};
