// @algolia/ui-library/public/components/CodeSample/CodeHighlighter/theme.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const colors = __importStar(require("../../../constants/colors"));
exports.default = {
    'code[class*="language-"]': {
        flex: 1,
        margin: '0',
        height: '100%',
        color: '#c5c8c6',
        textShadow: '0 1px rgba(0, 0, 0, 0.3)',
        fontFamily: "'JetBrains Mono', Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
        direction: 'ltr',
        textAlign: 'left',
        whiteSpace: 'pre',
        wordSpacing: 'normal',
        wordBreak: 'normal',
        lineHeight: '1.5',
        MozTabSize: '4',
        OTabSize: '4',
        tabSize: '4',
        WebkitHyphens: 'none',
        MozHyphens: 'none',
        msHyphens: 'none',
        hyphens: 'none',
    },
    'pre[class*="language-"]': {
        flex: 1,
        margin: '0',
        height: '100%',
        color: '#c5c8c6',
        textShadow: '0 1px rgba(0, 0, 0, 0.3)',
        fontFamily: "'JetBrains Mono', Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace",
        direction: 'ltr',
        textAlign: 'left',
        whiteSpace: 'pre',
        wordSpacing: 'normal',
        wordBreak: 'normal',
        lineHeight: '1.5',
        MozTabSize: '4',
        OTabSize: '4',
        tabSize: '4',
        WebkitHyphens: 'none',
        MozHyphens: 'none',
        msHyphens: 'none',
        hyphens: 'none',
        padding: '20px 20px 0px 20px',
        overflow: 'auto',
        background: 'transparent',
        fontSize: '14px',
    },
    ':not(pre) > code[class*="language-"]': {
        background: 'transparent',
        padding: '.1em',
        borderRadius: '.3em',
    },
    linenumber: {
        color: colors.GREY_700,
        borderRight: `solid 1px ${colors.GREY_700}`,
        marginRight: '16px',
        paddingRight: '16px',
        width: '50px',
    },
    comment: {
        color: '#7C7C7C',
    },
    prolog: {
        color: '#7C7C7C',
    },
    doctype: {
        color: '#7C7C7C',
    },
    cdata: {
        color: '#7C7C7C',
    },
    punctuation: {
        color: '#c5c8c6',
    },
    '.namespace': {
        Opacity: '.7',
    },
    property: {
        color: '#96CBFE',
    },
    keyword: {
        color: '#96CBFE',
    },
    tag: {
        color: '#96CBFE',
    },
    'class-name': {
        color: '#FFFFB6',
        textDecoration: 'underline',
    },
    boolean: {
        color: '#99CC99',
    },
    constant: {
        color: '#99CC99',
    },
    symbol: {
        color: '#f92672',
    },
    deleted: {
        color: '#f92672',
    },
    number: {
        color: '#FF73FD',
    },
    selector: {
        color: '#A8FF60',
    },
    'attr-name': {
        color: '#A8FF60',
    },
    string: {
        color: '#A8FF60',
    },
    char: {
        color: '#A8FF60',
    },
    builtin: {
        color: '#A8FF60',
    },
    inserted: {
        color: '#A8FF60',
    },
    variable: {
        color: '#C6C5FE',
    },
    operator: {
        color: '#EDEDED',
    },
    entity: {
        color: '#FFFFB6',
        cursor: 'help',
    },
    url: {
        color: '#96CBFE',
    },
    '.language-css .token.string': {
        color: '#87C38A',
    },
    '.style .token.string': {
        color: '#87C38A',
    },
    atrule: {
        color: '#F9EE98',
    },
    'attr-value': {
        color: '#F9EE98',
    },
    function: {
        color: '#DAD085',
    },
    regex: {
        color: '#E9C062',
    },
    important: {
        color: '#fd971f',
        fontWeight: 'bold',
    },
    bold: {
        fontWeight: 'bold',
    },
    italic: {
        fontStyle: 'italic',
    },
};
