// @algolia/ui-library/public/esm/layouts/TextBlock/TextBlock.js
/** @jsxRuntime classic /
/** @jsx jsx */
import { Children } from 'react';
import cx from 'classnames';
import { jsx } from '@emotion/react';
import { ErrorBoundary, Heading3, LabelText } from '../../index';
import styles from './TextBlock.css';
const COLORS = {
    blue: 'uil-color-nebula-500',
    green: 'uil-color-green-700',
    grey: 'uil-color-grey-800',
    orange: 'uil-color-orange-600',
    pink: 'uil-color-pink-600',
    red: 'uil-color-red-600',
    white: 'uil-color-white',
};
const TagWrapper = ({ item, index, isIconOutside }) => {
    if (!item)
        return null;
    const Tag = item.type || 'div';
    if (Tag.displayName) {
        if (Tag.displayName === 'Text' ||
            Tag.displayName === 'UserGeneratedContent' ||
            Tag.displayName === 'Card' ||
            Tag.displayName === 'Button') {
            return (jsx(Tag, Object.assign({}, item.props, { className: cx(item.props.className, index > 0 ? 'uil-mt-8' : 'uil-mt-0', isIconOutside && 'uil-gcstart-2', 'uil-mb-0') })));
        }
        else if (Tag.displayName === 'LightCta') {
            return null;
        }
        else {
            throw new Error("You can only use 'Text', 'Card', 'LightCta' or 'UserGeneratedContent' components.");
        }
    }
    else {
        throw new Error("You can only use 'Text', 'LightCta' or 'UserGeneratedContent' components.");
    }
};
const renderText = (children, isIconOutside) => Children.map(children, (item, index) => (jsx(ErrorBoundary, { key: index },
    jsx(TagWrapper, { item: item, index: index, isIconOutside: isIconOutside }))));
const renderCta = (children, color) => Children.map(children, item => item)
    .filter(item => item.type &&
    item.type.displayName &&
    item.type.displayName === 'LightCta')
    .map((item, index) => {
    const Tag = item.type || 'div';
    return (jsx("div", { key: index, className: "uil-mt-16 uil-mb-0" },
        jsx(Tag, Object.assign({ key: item }, item.props, { color: color, withTail: true }))));
});
/** `TextBlock` component is using different components like `Text` or `LightCta`.
 * It's made to be a block of text, with a title. Correct spacing is automatically handled around the elements within the component.
 *
 * It can be used in a parent component, on `TextsRow`.
 */
const TextBlock = ({ children, color = 'blue', icon, iconPosition = 'title', iconProps = {}, label, tag: CustomTag = 'article', title, titleProps = {}, withBigTitle = false, ...other }) => {
    const { className: titleClassName, ...otherTitleProps } = titleProps;
    const { className: iconClassName, ...otherIconProps } = iconProps;
    const { className: otherClassName, ...otherProps } = other;
    const CustomIconTag = icon;
    const isIconOutside = CustomIconTag && iconPosition === 'outside';
    const iconColor = COLORS[color];
    return (jsx(CustomTag, Object.assign({}, otherProps, { className: cx('md:uil-d-flex md:uil-fxd-column md:uil-jc-between', otherClassName) }),
        jsx("div", { css: isIconOutside ? styles.gridContainer : undefined, className: "uil-d-grid" },
            CustomIconTag && (iconPosition === 'above' || isIconOutside) && (jsx("div", { className: "uil-mb-8" },
                jsx(CustomIconTag, Object.assign({}, otherIconProps, { className: cx('uil-w-18 uil-h-18 lg:uil-w-24 lg:uil-h-24', iconClassName, iconColor) })))),
            !isIconOutside &&
                label &&
                (typeof label === 'string' ? (jsx(LabelText, { big: true, className: "uil-mt-0 uil-mb-8", color: color, iconProps: {
                        ...otherIconProps,
                        className: cx(iconColor, iconClassName),
                    }, icon: iconPosition === 'label' ? icon : undefined, tag: "p" }, label)) : (jsx("div", { className: "uil-mb-8" }, label))),
            title && (jsx(Heading3, Object.assign({}, otherTitleProps, { className: cx('uil-mt-0 uil-mb-8', titleClassName), iconProps: {
                    ...otherIconProps,
                    className: cx(iconColor, iconClassName, iconColor),
                }, icon: iconPosition === 'title' ? icon : undefined }), withBigTitle ? (jsx("span", { className: "uil-fsz-24 lg:uil-fsz-36" }, title)) : (title))),
            children && renderText(children, isIconOutside)),
        children && jsx("div", null, renderCta(children, color))));
};
TextBlock.displayName = 'TextBlock';
export default TextBlock;
