// @algolia/ui-library/public/esm/components/Card/Card.css.js
import { css } from '@emotion/react';
import * as breakpoints from '../../constants/breakpoints';
import * as colors from '../../constants/colors';
export default {
    root: css `
    transition: transform 0.2s ease-out, box-shadow 0.2s ease-out;
  `,
    light: css `
    background-image: linear-gradient(${colors.WHITE}, ${colors.GREY_100});
  `,
    dark: css `
    background-image: radial-gradient(
      circle at 50% 0,
      ${colors.GREY_700},
      ${colors.GREY_900}
    );
  `,
    blue: css `
    background-image: radial-gradient(
      circle at 50% 0,
      ${colors.NEBULA_500},
      ${colors.NEBULA_800}
    );
  `,
    green: css `
    background-image: radial-gradient(
      circle at 50% 0,
      ${colors.GREEN_700},
      ${colors.GREEN_900}
    );
  `,
    elevation: css `
    &:hover,
    &:focus {
      transform: translateY(-5px);
      box-shadow: 0 10px 30px 0 rgba(37, 44, 97, 0.15),
        0 4px 18px 0 rgba(93, 100, 148, 0.2);
    }
  `,
    largeElevation: css `
    &:hover,
    &:focus {
      transform: translateY(-5px);
      box-shadow: 0 20px 60px 0 rgba(37, 44, 97, 0.15),
        0 8px 16px 0 rgba(93, 100, 148, 0.2);
    }
  `,
    badge: css `
    @media (min-width: ${breakpoints.SM}) {
      left: 50%;
      transform: translateX(-50%) translateY(-50%);
    }
  `,
};
