// @algolia/ui-library/public/components/ErrorBoundary.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ErrorBoundary = void 0;
const react_1 = __importDefault(require("react"));
const classnames_1 = __importDefault(require("classnames"));
const Error_1 = __importDefault(require("./Error"));
class ErrorBoundary extends react_1.default.Component {
    constructor() {
        super(...arguments);
        this.state = {
            hasError: false,
            errorMessage: '',
        };
    }
    static getDerivedStateFromError(error) {
        return {
            hasError: true,
            errorMessage: error.message,
        };
    }
    render() {
        const { className, children } = this.props;
        const { hasError, errorMessage } = this.state;
        if (hasError) {
            return react_1.default.createElement(Error_1.default, { className: classnames_1.default(className), text: errorMessage });
        }
        return children;
    }
}
exports.ErrorBoundary = ErrorBoundary;
exports.default = ErrorBoundary;
