// @algolia/ui-library/public/components/Nav/Nav.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const cloudinary_1 = __importDefault(require("../../constants/cloudinary"));
const breakpoints = __importStar(require("../../constants/breakpoints"));
const colors = __importStar(require("../../constants/colors"));
exports.default = {
    desktopNavShadow: (theme) => {
        const boxShadowColor = theme === 'dark' ? 'rgb(0 0 0 / 2.5%)' : 'rgb(55 58 80 / 5%)';
        return react_1.css `
      box-shadow: 0px 16px 32px ${boxShadowColor},
        0px 8px 16px ${boxShadowColor}, 0px 4px 8px ${boxShadowColor},
        0px 2px 4px ${boxShadowColor}, 0px 1px 2px ${boxShadowColor};
    `;
    },
    desktopNavLinkBefore: (isCurrent, theme) => react_1.css `
    &:before {
      content: attr(data-title);
      position: absolute;
      text-align: center;
      margin: auto;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      font-family: Poppins, Arial, sans-serif;
      font-weight: 600;
      letter-spacing: normal;
      white-space: nowrap;
      color: ${theme === 'light' ? colors.GREY_900 : colors.WHITE};
      z-index: 2;
      display: ${isCurrent ? 'block' : 'none'};
      pointer-events: none;
      font-size: 16px;

      @media (min-width: ${breakpoints.LG}) {
        transform: translate(-50%, calc(-50% - 1px));
      }
    }
  `,
    subNavLinks: react_1.css `
    &:before,
    &:after {
      content: '';
      position: fixed;
      left: 0;
      right: 0;
      height: 24px;
      pointer-events: none;
      @media (min-width: ${breakpoints.MD}) {
        display: none;
      }
    }
    &:before {
      background-image: linear-gradient(
        rgb(255 255 255 / 100%),
        rgb(255 255 255 / 0%)
      );
    }
    &:after {
      bottom: 70px;
      background-image: linear-gradient(
        to top,
        rgb(255 255 255 / 100%),
        rgb(255 255 255 / 0%)
      );
    }
  `,
    subNavItem: react_1.css `
    transition: opacity 0.2s ease-out, transform 0.2s ease-out;
    @media (min-width: ${breakpoints.MD}) {
      transform: translateY(-10px);
      margin-right: 16px;
      &:hover,
      &:focus {
        transform: translateY(0);
      }
    }
  `,
    subNavItemProducts: react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      width: auto;
    }
  `,
    subNavItemFocused: react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      transform: translateY(0);
    }
  `,
    subNavBannerBackground: react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      background-image: url('${cloudinary_1.default}/nav/search-inspiration-banner.png');
      background-position: 125%
    }
  `,
    subNavBannerLink: (theme, backgroundImage) => theme === 'dark'
        ? react_1.css `
          background-image: radial-gradient(100% 100% at 100% 0%, ${colors.GREY_700} 0%, ${colors.GREY_900} 100%);
          @media (min-width: ${breakpoints.MD}) {
            background-image: url('${cloudinary_1.default}${backgroundImage}'),
              radial-gradient(100% 100% at 100% 0%, ${colors.GREY_700} 0%, ${colors.GREY_900} 100%)
          }
        `
        : react_1.css `
          @media (min-width: ${breakpoints.MD}) {
            background-image: url('${cloudinary_1.default}${backgroundImage}')
          }
      `,
    navSearch: react_1.css `
    height: 48px;
  `,
    navSearchTranslucide: react_1.css `
    background-color: rgb(72 76 122 / 50%);
  `,
    navSearchIcon: react_1.css `
    top: -2px;
  `,
    navMobileSearchIcon: react_1.css `
    top: -2px;
    transition: none !important;
  `,
    secondaryNav: (theme, hasGradient) => {
        const boxShadowColor = theme === 'dark' ? 'rgb(0 0 0 / 2.5%)' : 'rgb(55 58 80 / 5%)';
        const backgroundColor = (opacity) => {
            return theme === 'dark'
                ? `rgb(35 38 59 / ${opacity})`
                : `rgb(245 245 250 / ${opacity})`;
        };
        return react_1.css `
      transition: none !important;
      box-shadow: 0px 16px 32px ${boxShadowColor},
        0px 8px 16px ${boxShadowColor}, 0px 4px 8px ${boxShadowColor},
        0px 2px 4px ${boxShadowColor}, 0px 1px 2px ${boxShadowColor};
      background: ${backgroundColor('97%')};

      @supports (backdrop-filter: blur(5px)) {
        backdrop-filter: blur(5px);
        background: ${backgroundColor('85%')};
      }

      ${hasGradient &&
            react_1.css `
          &::before,
          &::after {
            content: '';
            display: block;
            position: absolute;
            top: 0;
            left: 0;
            width: 16px;
            height: 100%;
            background: linear-gradient(
              to right,
              ${theme === 'dark' ? colors.GREY_900 : colors.GREY_100},
              rgb(0 0 0 / 0)
            );
            z-index: 3;
          }
          :after {
            width: 100px;
            left: calc(100vw - 100px) !important;
            background: linear-gradient(
              to left,
              ${theme === 'dark' ? colors.GREY_900 : colors.GREY_100},
              ${theme === 'dark' ? colors.GREY_900 : colors.GREY_100},
              rgba(0 0 0 / 0)
            );
          }
        `}
    `;
    },
    secondaryNavInner: react_1.css `
    max-width: calc(${breakpoints.XL} + 40px);
    -webkit-overflow-scrolling: touch;
    -ms-overflow-style: -ms-autohiding-scrollbar;
  `,
};
