// @algolia/ui-library/public/components/Nav/SubNav.data.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const routes = __importStar(require("../../constants/routes"));
const externalRoutes = __importStar(require("../../constants/external-routes"));
exports.default = [
    {
        title: 'Products',
        columns: 2,
        links: [
            {
                title: 'Search',
                description: 'Powerful, hosted search API to create fast and relevant search & discovery.',
                icon: '/nav/search.svg',
                url: routes.HOSTED_SEARCH_API,
                cta: {
                    text: 'Learn more',
                },
            },
            {
                title: 'Recommend',
                description: 'Flexible, hosted recommendation API with advanced programmatic control.',
                icon: '/nav/recommend.svg',
                url: routes.RECOMMENDATIONS,
                cta: {
                    text: 'Learn more',
                    color: 'red',
                },
            },
        ],
    },
    {
        title: 'Solutions & Industries',
        columnTitles: ['Solutions', 'Industries'],
        links: [
            {
                title: 'Salesforce Commerce Cloud',
                icon: '/icons/salesforce.svg',
                url: routes.SALESFORCE_COMMERCE_CLOUD,
            },
            {
                title: 'Shopify',
                icon: '/icons/shopify.svg',
                url: routes.SHOPIFY,
            },
            {
                title: 'Magento (Adobe)',
                icon: '/icons/magento.svg',
                url: routes.MAGENTO,
            },
            {
                title: 'Netlify',
                icon: '/icons/netlify.svg',
                url: routes.NETLIFY,
            },
            {
                title: 'Contentful',
                icon: '/icons/contentful.svg',
                url: routes.CONTENTFUL_SEARCH,
            },
        ],
        sidebar: [
            {
                title: 'Ecommerce',
                url: routes.ECOMMERCE,
            },
            {
                title: 'B2B Ecommerce',
                url: routes.B2B_ECOMMERCE,
            },
            {
                title: 'Marketplaces',
                url: routes.MARKETPLACE_SEARCH,
            },
            {
                title: 'Media',
                url: routes.MEDIA,
            },
            {
                title: 'SaaS',
                url: routes.SAAS,
            },
            {
                title: 'Inspiration Library',
                url: routes.SEARCH_INSPIRATION_LIBRARY,
                mobileOnly: true,
                dismissHover: true,
            },
        ],
        banner: {
            title: 'Inspiration Library',
            description: 'Get inspired by 200+ customer examples and take your search and discovery experience to the next level.',
            url: routes.SEARCH_INSPIRATION_LIBRARY,
            hideOnMobile: true,
        },
    },
    {
        title: 'Developers',
        links: [
            {
                title: 'Developer hub',
                url: routes.DEVELOPERS,
                color: 'white',
                theme: 'dark',
                backgroundImage: '/nav/terminal.png',
            },
            {
                title: 'Documentations',
                urls: [
                    {
                        title: 'API clients and integrations',
                        url: routes.INTEGRATIONS,
                        fullWidth: true,
                    },
                    {
                        title: 'Docs',
                        url: externalRoutes.DOC,
                    },
                    {
                        title: 'Get started',
                        url: routes.DEVELOPERS_GET_STARTED,
                    },
                ],
            },
            {
                title: 'Resources',
                urls: [
                    {
                        title: 'For Open Source',
                        url: routes.FOR_OPEN_SOURCE,
                    },
                    {
                        title: 'Discourse',
                        url: externalRoutes.DISCOURSE,
                    },
                    {
                        title: 'Github',
                        url: externalRoutes.GITHUB,
                    },
                ],
            },
        ],
    },
    {
        title: 'Pricing',
        links: [
            {
                title: 'Pricing',
                url: routes.PRICING,
            },
            {
                title: 'Calculator',
                url: `${routes.PRICING}#calculator`,
            },
            {
                title: 'FAQs',
                url: `${routes.PRICING}#faq`,
            },
        ],
    },
    {
        title: 'Resources',
        columnTitles: ['Customers', 'Resources'],
        links: [
            {
                title: 'Inspiration Library',
                url: routes.SEARCH_INSPIRATION_LIBRARY,
            },
            {
                title: 'Our Customers',
                url: routes.CUSTOMERS,
            },
            {
                title: 'Customer Stories',
                url: externalRoutes.RESOURCES_CASE_STUDIES,
            },
            {
                title: 'Algolia Academy',
                url: externalRoutes.ACADEMY,
            },
            {
                title: 'Contact Support',
                url: routes.CONTACT_SUPPORT,
            },
        ],
        sidebar: [
            {
                title: 'Resource Center',
                url: externalRoutes.RESOURCES,
            },
            {
                title: 'Blog',
                url: externalRoutes.BLOG,
            },
            {
                title: 'Events',
                url: routes.EVENTS,
            },
            {
                title: 'Grade your search',
                url: externalRoutes.GRADER,
            },
        ],
    },
];
