// @algolia/ui-library/public/esm/components/Carousel/Carousel.css.js
import { css } from '@emotion/react';
import * as breakpoints from '../../constants/breakpoints';
export default {
    slide: css `
    transition: transform 0.2s ease-out, z-index 0.2s ease-out;
    transform-origin: center;
  `,
    getLeftSlide: (isWide) => css `
    @media (min-width: ${breakpoints.MD}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-87%' : '-116%'});
      opacity: 0.3;
    }
    @media (min-width: ${breakpoints.LG}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-87%' : '-149%'});
    }
  `,
    middleSlide: css `
    @media (min-width: ${breakpoints.MD}) {
      transform: translateX(-50%);
      z-index: 1;
    }
  `,
    getRightSlide: (isWide) => css `
    @media (min-width: ${breakpoints.MD}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-38%' : '-17%'});
      opacity: 0.3;
    }
    @media (min-width: ${breakpoints.LG}) {
      transform: scale(${isWide ? '0.8' : '0.75'})
        translateX(${isWide ? '-38%' : '16%'});
    }
  `,
    buttonLeft: css `
    @media (min-width: ${breakpoints.MD}) {
      right: calc(100% + 20px);
      transform: translate(0, -50%);
      transition: transform 0.2s ease-in-out;
      &:hover {
        transform: translate(-2px, -50%);
      }
    }
  `,
    buttonRight: css `
    @media (min-width: ${breakpoints.MD}) {
      left: calc(100% + 20px);
      transform: translate(0, -50%);
      transition: transform 0.2s ease-in-out;
      &:hover {
        transform: translate(2px, -50%);
      }
    }
  `,
    dotsContainer: css `
    bottom: -48px;
  `,
};
