// @algolia/ui-library/public/constants/colors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.COLOR_RECOMMEND = exports.COLOR_ANSWER = exports.COLOR_PREDICT = exports.GREEN_900 = exports.GREEN_800 = exports.GREEN_700 = exports.GREEN_600 = exports.GREEN_500 = exports.GREEN_400 = exports.GREEN_300 = exports.GREEN_200 = exports.GREEN_100 = exports.CYAN_900 = exports.CYAN_800 = exports.CYAN_700 = exports.CYAN_600 = exports.CYAN_500 = exports.CYAN_400 = exports.CYAN_300 = exports.CYAN_200 = exports.CYAN_100 = exports.RED_900 = exports.RED_800 = exports.RED_700 = exports.RED_600 = exports.RED_500 = exports.RED_400 = exports.RED_300 = exports.RED_200 = exports.RED_100 = exports.PINK_900 = exports.PINK_800 = exports.PINK_700 = exports.PINK_600 = exports.PINK_500 = exports.PINK_400 = exports.PINK_300 = exports.PINK_200 = exports.PINK_100 = exports.NEBULA_900 = exports.NEBULA_800 = exports.NEBULA_700 = exports.NEBULA_600 = exports.NEBULA_500 = exports.NEBULA_400 = exports.NEBULA_300 = exports.NEBULA_200 = exports.NEBULA_100 = exports.ORANGE_900 = exports.ORANGE_800 = exports.ORANGE_700 = exports.ORANGE_600 = exports.ORANGE_500 = exports.ORANGE_400 = exports.ORANGE_300 = exports.ORANGE_200 = exports.ORANGE_100 = exports.GREY_900 = exports.GREY_800 = exports.GREY_700 = exports.GREY_600 = exports.GREY_500 = exports.GREY_400 = exports.GREY_300 = exports.GREY_200 = exports.GREY_100 = exports.GREY_050 = exports.TRANSPARENT = exports.WHITE = void 0;
const config = require('../fragments/fragments');
exports.WHITE = config.colors.white;
exports.TRANSPARENT = config.colors.transparent;
// New synapse colors
exports.GREY_050 = config.colors['grey-050'];
exports.GREY_100 = config.colors['grey-100'];
exports.GREY_200 = config.colors['grey-200'];
exports.GREY_300 = config.colors['grey-300'];
exports.GREY_400 = config.colors['grey-400'];
exports.GREY_500 = config.colors['grey-500'];
exports.GREY_600 = config.colors['grey-600'];
exports.GREY_700 = config.colors['grey-700'];
exports.GREY_800 = config.colors['grey-800'];
exports.GREY_900 = config.colors['grey-900'];
exports.ORANGE_100 = config.colors['orange-100'];
exports.ORANGE_200 = config.colors['orange-200'];
exports.ORANGE_300 = config.colors['orange-300'];
exports.ORANGE_400 = config.colors['orange-400'];
exports.ORANGE_500 = config.colors['orange-500'];
exports.ORANGE_600 = config.colors['orange-600'];
exports.ORANGE_700 = config.colors['orange-700'];
exports.ORANGE_800 = config.colors['orange-800'];
exports.ORANGE_900 = config.colors['orange-900'];
exports.NEBULA_100 = config.colors['nebula-100'];
exports.NEBULA_200 = config.colors['nebula-200'];
exports.NEBULA_300 = config.colors['nebula-300'];
exports.NEBULA_400 = config.colors['nebula-400'];
exports.NEBULA_500 = config.colors['nebula-500'];
exports.NEBULA_600 = config.colors['nebula-600'];
exports.NEBULA_700 = config.colors['nebula-700'];
exports.NEBULA_800 = config.colors['nebula-800'];
exports.NEBULA_900 = config.colors['nebula-900'];
exports.PINK_100 = config.colors['pink-100'];
exports.PINK_200 = config.colors['pink-200'];
exports.PINK_300 = config.colors['pink-300'];
exports.PINK_400 = config.colors['pink-400'];
exports.PINK_500 = config.colors['pink-500'];
exports.PINK_600 = config.colors['pink-600'];
exports.PINK_700 = config.colors['pink-700'];
exports.PINK_800 = config.colors['pink-800'];
exports.PINK_900 = config.colors['pink-900'];
exports.RED_100 = config.colors['red-100'];
exports.RED_200 = config.colors['red-200'];
exports.RED_300 = config.colors['red-300'];
exports.RED_400 = config.colors['red-400'];
exports.RED_500 = config.colors['red-500'];
exports.RED_600 = config.colors['red-600'];
exports.RED_700 = config.colors['red-700'];
exports.RED_800 = config.colors['red-800'];
exports.RED_900 = config.colors['red-900'];
exports.CYAN_100 = config.colors['cyan-100'];
exports.CYAN_200 = config.colors['cyan-200'];
exports.CYAN_300 = config.colors['cyan-300'];
exports.CYAN_400 = config.colors['cyan-400'];
exports.CYAN_500 = config.colors['cyan-500'];
exports.CYAN_600 = config.colors['cyan-600'];
exports.CYAN_700 = config.colors['cyan-700'];
exports.CYAN_800 = config.colors['cyan-800'];
exports.CYAN_900 = config.colors['cyan-900'];
exports.GREEN_100 = config.colors['green-100'];
exports.GREEN_200 = config.colors['green-200'];
exports.GREEN_300 = config.colors['green-300'];
exports.GREEN_400 = config.colors['green-400'];
exports.GREEN_500 = config.colors['green-500'];
exports.GREEN_600 = config.colors['green-600'];
exports.GREEN_700 = config.colors['green-700'];
exports.GREEN_800 = config.colors['green-800'];
exports.GREEN_900 = config.colors['green-900'];
exports.COLOR_PREDICT = config.colors['color-predict'];
exports.COLOR_ANSWER = config.colors['color-answer'];
exports.COLOR_RECOMMEND = config.colors['color-recommend'];
