// @algolia/ui-library/public/constants/external-routes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WEBINAR_WINTER_20 = exports.WEBINAR_SUMMER_19 = exports.ZENDESK_GET_STARTED = exports.YOUTUBE = exports.TECH_BLOG = exports.TALKSEARCH = exports.SUPPORT = exports.STORIES = exports.STATUS = exports.SIGN_UP = exports.SIGN_OUT = exports.SIGN_IN = exports.SHOW_AND_TELL = exports.RESOURCES_CASE_STUDIES = exports.RESOURCES = exports.PARTNERS = exports.NEWSLETTER = exports.GRADER = exports.GITHUB = exports.GETTING_STARTED = exports.FAQ = exports.DOCSEARCH = exports.DISCOURSE = exports.SEARCH_AND_DISCOVERY = exports.DOC_CHANGELOG = exports.DOC = exports.DASHBOARD = exports.COMMUNITY = exports.BLOG = exports.ACADEMY = void 0;
exports.ACADEMY = 'https://academy.algolia.com/';
exports.BLOG = '/blog/';
exports.COMMUNITY = 'https://community.algolia.com';
exports.DASHBOARD = '/dashboard';
exports.DOC = '/doc';
exports.DOC_CHANGELOG = `${exports.DOC}/changelog/`;
exports.SEARCH_AND_DISCOVERY = `${exports.DOC}/guides/solutions/ecommerce/ui-kits/`;
exports.DISCOURSE = 'https://discourse.algolia.com';
exports.DOCSEARCH = 'https://docsearch.algolia.com';
exports.FAQ = `${exports.DOC}/faq/`;
exports.GETTING_STARTED = `${exports.DOC}/guides/getting-started/how-algolia-works/`;
exports.GITHUB = 'https://github.com/algolia';
exports.GRADER = 'https://grader.algolia.com';
exports.NEWSLETTER = 'https://go.algolia.com/subscription_newsletter/';
exports.PARTNERS = 'https://partners.algolia.com';
exports.RESOURCES = 'https://resources.algolia.com';
exports.RESOURCES_CASE_STUDIES = `${exports.RESOURCES}/customer-stories`;
exports.SHOW_AND_TELL = 'https://discourse.algolia.com/c/show-tell';
exports.SIGN_IN = '/users/sign_in';
exports.SIGN_OUT = '/users/sign_out';
exports.SIGN_UP = '/users/sign_up';
exports.STATUS = 'https://status.algolia.com';
exports.STORIES = 'https://stories.algolia.com';
exports.SUPPORT = '/support/';
exports.TALKSEARCH = 'https://community.algolia.com/talksearch/';
exports.TECH_BLOG = '/tech-blog/';
exports.YOUTUBE = 'https://www.youtube.com/user/Algolia';
exports.ZENDESK_GET_STARTED = '/getstarted#/zendesk';
// Temporary constants, remove when not used anymore
exports.WEBINAR_SUMMER_19 = '/webinar/summer-19/';
exports.WEBINAR_WINTER_20 = 'https://go.algolia.com/webinar-winter-20/';
