// @algolia/ui-library/public/constants/routes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CHECK_EMAIL_EMBED = exports.BUSINESS_INFORMATION_EMBED = exports.ABOUT_YOU_EMBED = exports.SIGN_UP_EMBED = exports.WHY = exports.TECH_BLOG = exports.STATE_OF_SEARCH = exports.SOCIAL_IMPACT = exports.SEARCH_INSPIRATION_LIBRARY = exports.SEARCH = exports.REDEEM = exports.PRODUCTS = exports.PRICING = exports.PRESS = exports.INTEGRATIONS = exports.HOME = exports.FOR_OPEN_SOURCE = exports.EVENTS = exports.ERROR = exports.CONTENTFUL_SEARCH = exports.DEVELOPERS_GET_STARTED = exports.DEVELOPERS = exports.DEMO_REQUEST = exports.COVID_19 = exports.CONTACT_SUPPORT = exports.CONTACT_US = exports.COMPARE_ALGOLIA_VS_ELASTICSEARCH = exports.CAREERS = exports.ALGOLIA_VS_ELASTICSEARCH = exports.ABOUT = exports.CUSTOMERS = exports.SHOPIFY = exports.SALESFORCE_COMMERCE_CLOUD = exports.SECURITY_MEASURES = exports.INFRASTRUCTURE = exports.SECURITY = exports.SAAS = exports.NETLIFY = exports.MEDIA = exports.MARKETPLACE_SEARCH = exports.MAGENTO = exports.GEO_SEARCH = exports.ECOMMERCE = exports.B2B_ECOMMERCE = exports.AZURE = exports.SUMMER_19 = exports.WHATS_NEW = exports.RECOMMENDATIONS = exports.CRAWLER = exports.AI_STUDIO = exports.DYNAMIC_RERANKING = exports.DYNAMIC_SYNONYM_SUGGESTIONS = exports.AI_SEARCH = exports.PRODUCTS_SEARCH = exports.VISUAL_MERCHANDISING_CURATION = exports.VOICE_SEARCH = exports.UI_COMPONENT_LIBRARIES = exports.SITE_SEARCH = exports.SEARCH_RELEVANCE_RULES = exports.PERSONALIZED_QUERY_SUGGESTIONS = exports.PERSONALIZATION = exports.MOBILE_SEARCH = exports.HOSTED_SEARCH_API = exports.HEADLESS_COMMERCE = exports.ENTERPRISE = exports.ANSWERS_FOR_PUBLISHERS = exports.ANSWERS_FOR_CUSTOMER_SUPPORT = exports.ANSWERS = exports.ANALYTICS = exports.AB_TESTING = exports.TERMS = exports.POLICIES_SUPPORT = exports.SLA = exports.PRIVACY = exports.POLICIES_OLD = exports.LEGAL = exports.INFRASTRUCTURE_AND_SUBPROCESSORS = exports.COOKIES = exports.ACCEPTABLE_USE = exports.ZENDESK = void 0;
exports.ZENDESK = '/integrations/zendesk/';
exports.ACCEPTABLE_USE = '/policies/acceptable-use/';
exports.COOKIES = '/policies/cookies/';
exports.INFRASTRUCTURE_AND_SUBPROCESSORS = '/policies/infrastructure-and-sub-processors/';
exports.LEGAL = '/policies/legal/';
exports.POLICIES_OLD = '/policies/old/';
exports.PRIVACY = '/policies/privacy/';
exports.SLA = '/policies/sla/';
exports.POLICIES_SUPPORT = '/policies/support/';
exports.TERMS = '/policies/terms/';
const SEARCH_AND_DISCOVERY = '/products/search-and-discovery/';
const DISTRIBUTED_SECURE = '/distributed-secure/';
exports.AB_TESTING = `${SEARCH_AND_DISCOVERY}ab-testing/`;
exports.ANALYTICS = `${SEARCH_AND_DISCOVERY}analytics/`;
exports.ANSWERS = `${SEARCH_AND_DISCOVERY}answers/`;
exports.ANSWERS_FOR_CUSTOMER_SUPPORT = `${SEARCH_AND_DISCOVERY}answers-for-customer-support/`;
exports.ANSWERS_FOR_PUBLISHERS = `${SEARCH_AND_DISCOVERY}answers-for-publishers/`;
exports.ENTERPRISE = `${SEARCH_AND_DISCOVERY}enterprise/`;
exports.HEADLESS_COMMERCE = `${SEARCH_AND_DISCOVERY}headless-commerce/`;
exports.HOSTED_SEARCH_API = `${SEARCH_AND_DISCOVERY}hosted-search-api/`;
exports.MOBILE_SEARCH = `${SEARCH_AND_DISCOVERY}mobile-search/`;
exports.PERSONALIZATION = `${SEARCH_AND_DISCOVERY}personalization/`;
exports.PERSONALIZED_QUERY_SUGGESTIONS = `${SEARCH_AND_DISCOVERY}personalized-query-suggestions/`;
exports.SEARCH_RELEVANCE_RULES = `${SEARCH_AND_DISCOVERY}search-relevance-rules/`;
exports.SITE_SEARCH = `${SEARCH_AND_DISCOVERY}site-search/`;
exports.UI_COMPONENT_LIBRARIES = `${SEARCH_AND_DISCOVERY}ui-component-libraries/`;
exports.VOICE_SEARCH = `${SEARCH_AND_DISCOVERY}voice-search/`;
exports.VISUAL_MERCHANDISING_CURATION = `${SEARCH_AND_DISCOVERY}visual-merchandising-curation/`;
exports.PRODUCTS_SEARCH = `${SEARCH_AND_DISCOVERY}hosted-search-api/`;
exports.AI_SEARCH = `/products/ai-search/`;
exports.DYNAMIC_SYNONYM_SUGGESTIONS = `${exports.AI_SEARCH}dynamic-synonym-suggestions/`;
exports.DYNAMIC_RERANKING = `${exports.AI_SEARCH}dynamic-reranking/`;
exports.AI_STUDIO = `/products/ai-studio/`;
exports.CRAWLER = `/products/crawler/`;
exports.RECOMMENDATIONS = `/products/recommendations/`;
exports.WHATS_NEW = `/products/whats-new/`;
exports.SUMMER_19 = `/products/whats-new/summer19/`;
const SEARCH_SOLUTIONS = '/search-solutions/';
exports.AZURE = `${SEARCH_SOLUTIONS}algolia-on-azure/`;
exports.B2B_ECOMMERCE = `${SEARCH_SOLUTIONS}b2b-ecommerce/`;
exports.ECOMMERCE = `${SEARCH_SOLUTIONS}ecommerce/`;
exports.GEO_SEARCH = `${SEARCH_SOLUTIONS}geo-search/`;
exports.MAGENTO = `${SEARCH_SOLUTIONS}magento/`;
exports.MARKETPLACE_SEARCH = `${SEARCH_SOLUTIONS}marketplaces/`;
exports.MEDIA = `${SEARCH_SOLUTIONS}media/`;
exports.NETLIFY = `${SEARCH_SOLUTIONS}netlify/`;
exports.SAAS = `${SEARCH_SOLUTIONS}saas/`;
exports.SECURITY = `${DISTRIBUTED_SECURE}security-compliance/`;
exports.INFRASTRUCTURE = `${DISTRIBUTED_SECURE}global-infrastructure/`;
exports.SECURITY_MEASURES = `${exports.SECURITY}/measures/`;
exports.SALESFORCE_COMMERCE_CLOUD = `${SEARCH_SOLUTIONS}salesforce-commerce-cloud/`;
exports.SHOPIFY = `${SEARCH_SOLUTIONS}shopify/`;
exports.CUSTOMERS = '/resources/customers/';
exports.ABOUT = '/about/';
exports.ALGOLIA_VS_ELASTICSEARCH = '/algolia-vs-elasticsearch/';
exports.CAREERS = '/careers/';
exports.COMPARE_ALGOLIA_VS_ELASTICSEARCH = '/compare-algolia-vs-elasticsearch/';
exports.CONTACT_US = '/contactus/';
exports.CONTACT_SUPPORT = '/support/?contact=';
exports.COVID_19 = '/covid-19/';
exports.DEMO_REQUEST = '/demorequest/';
exports.DEVELOPERS = '/developers/';
exports.DEVELOPERS_GET_STARTED = `${exports.DEVELOPERS}get-started`;
exports.CONTENTFUL_SEARCH = `${exports.DEVELOPERS}contentful-search-algolia`;
exports.ERROR = '404';
exports.EVENTS = '/events/';
exports.FOR_OPEN_SOURCE = '/for-open-source/';
exports.HOME = '/';
exports.INTEGRATIONS = `${exports.DEVELOPERS}#integrations`;
exports.PRESS = '/press/';
exports.PRICING = '/pricing/';
exports.PRODUCTS = '/products/';
exports.REDEEM = '/redeem/';
exports.SEARCH = '/search/';
exports.SEARCH_INSPIRATION_LIBRARY = '/search-inspiration-library/';
exports.SOCIAL_IMPACT = '/social-impact/';
exports.STATE_OF_SEARCH = '/state-of-site-search-and-discovery/';
exports.TECH_BLOG = '/tech-blog/';
exports.WHY = '/why/';
exports.SIGN_UP_EMBED = '/embed/sign-up/';
exports.ABOUT_YOU_EMBED = '/embed/about-you/';
exports.BUSINESS_INFORMATION_EMBED = '/embed/business-information/';
exports.CHECK_EMAIL_EMBED = '/embed/check-email/';
