// @algolia/ui-library/public/components/Switch/Switch.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const colors = __importStar(require("../../constants/colors"));
exports.default = {
    input: react_1.css `
    &:checked + div::before {
      transform: translateX(calc(100% + 4px));
    }
    & + div {
      background-color: ${colors.GREY_900};
      &:before {
        background-color: ${colors.NEBULA_500};
      }
    }
  `,
    toggleTransition: react_1.css `
    transition: background-color 0.2s ease-in-out;
    &::before {
      content: '';
      width: 24px;
      height: 24px;
      display: block;
      border-radius: 100%;
      transition: transform 0.2s ease-in-out, background-color 0.2s ease-in-out;
      transform: none;
      box-shadow: inset 0 1px 0 0 rgba(255, 255, 255, 0.5);
    }
  `,
};
