// @algolia/ui-library/public/esm/components/Pill/Pill.js
import cx from 'classnames';
import { jsx } from '@emotion/react';
import { LabelText } from '../../index';
import styles from './Pill.css';
const BACKGROUND_COLORS = {
    blue: 'uil-bgc-nebula-200',
    green: 'uil-bgc-green-200',
    grey: 'uil-bgc-grey-200',
    orange: 'uil-bgc-orange-200',
    pink: 'uil-bgc-pink-200',
};
const CURSORS = {
    auto: 'uil-cursor-auto',
    pointer: 'uil-cursor-pointer',
    'not-allowed': 'uil-cursor-not-allowed',
};
/** As an example, you can use a `Pill` if you want to show filterable options. */
const Pill = ({ active = false, children, color = 'blue', cursor = 'pointer', rounded = true, small = false, tag: CustomTag = 'button', ...other }) => {
    const { className: otherClassName, css: otherCss, ...otherProps } = other;
    return (jsx(CustomTag, Object.assign({}, otherProps, { css: [!small && styles.root, otherCss], className: cx(`uil-d-inline-block uil-app-none uil-bdw-0 uil-ph-12 uil-h-100p`, CURSORS[cursor], active ? BACKGROUND_COLORS[color] : 'uil-bgc-grey-200', small && 'uil-pt-0 uil-pb-0', rounded ? 'uil-bdr-max' : 'uil-bdr-4', otherClassName) }),
        jsx(LabelText, { color: active ? color : 'grey', className: "uil-pe-none uil-pos-relative" }, children)));
};
export default Pill;
