// @algolia/ui-library/public/components/Pill/Pill.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const classnames_1 = __importDefault(require("classnames"));
const react_1 = require("@emotion/react");
const index_1 = require("../../index");
const Pill_css_1 = __importDefault(require("./Pill.css"));
const BACKGROUND_COLORS = {
    blue: 'uil-bgc-nebula-200',
    green: 'uil-bgc-green-200',
    grey: 'uil-bgc-grey-200',
    orange: 'uil-bgc-orange-200',
    pink: 'uil-bgc-pink-200',
};
const CURSORS = {
    auto: 'uil-cursor-auto',
    pointer: 'uil-cursor-pointer',
    'not-allowed': 'uil-cursor-not-allowed',
};
/** As an example, you can use a `Pill` if you want to show filterable options. */
const Pill = ({ active = false, children, color = 'blue', cursor = 'pointer', rounded = true, small = false, tag: CustomTag = 'button', ...other }) => {
    const { className: otherClassName, css: otherCss, ...otherProps } = other;
    return (react_1.jsx(CustomTag, Object.assign({}, otherProps, { css: [!small && Pill_css_1.default.root, otherCss], className: classnames_1.default(`uil-d-inline-block uil-app-none uil-bdw-0 uil-ph-12 uil-h-100p`, CURSORS[cursor], active ? BACKGROUND_COLORS[color] : 'uil-bgc-grey-200', small && 'uil-pt-0 uil-pb-0', rounded ? 'uil-bdr-max' : 'uil-bdr-4', otherClassName) }),
        react_1.jsx(index_1.LabelText, { color: active ? color : 'grey', className: "uil-pe-none uil-pos-relative" }, children)));
};
exports.default = Pill;
