// @algolia/fragments.js/lib/modules/backgrounds/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "background", {
    enumerable: true,
    get: function() {
        return _background.default;
    }
});
Object.defineProperty(exports, "backgroundAttachment", {
    enumerable: true,
    get: function() {
        return _backgroundAttachment.default;
    }
});
Object.defineProperty(exports, "backgroundColor", {
    enumerable: true,
    get: function() {
        return _backgroundColor.default;
    }
});
Object.defineProperty(exports, "backgroundPosition", {
    enumerable: true,
    get: function() {
        return _backgroundPosition.default;
    }
});
Object.defineProperty(exports, "backgroundRepeat", {
    enumerable: true,
    get: function() {
        return _backgroundRepeat.default;
    }
});
Object.defineProperty(exports, "backgroundSize", {
    enumerable: true,
    get: function() {
        return _backgroundSize.default;
    }
});
var _background = _interopRequireDefault(require("./background"));
var _backgroundAttachment = _interopRequireDefault(require("./background-attachment"));
var _backgroundColor = _interopRequireDefault(require("./background-color"));
var _backgroundPosition = _interopRequireDefault(require("./background-position"));
var _backgroundRepeat = _interopRequireDefault(require("./background-repeat"));
var _backgroundSize = _interopRequireDefault(require("./background-size"));
function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
