// @algolia/ui-library/public/components/KeyNumberCard/KeyNumberCard.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const breakpoints = __importStar(require("../../constants/breakpoints"));
const colors = __importStar(require("../../constants/colors"));
exports.default = {
    logoBlock: react_1.css `
    &:after {
      content: '';
      display: block;
      position: absolute;
      bottom: 0;
      background: ${colors.GREY_200};
      height: 1px;
      width: 100%;
    }
    @media (min-width: ${breakpoints.MD}) {
      order: 0;
      &:after {
        content: '';
        display: block;
        position: absolute;
        top: 0;
        left: 100%;
        bottom: 0;
        background: ${colors.GREY_200};
        height: 100%;
        width: 1px;
      }
    }
  `,
    customerLogo: react_1.css `
    max-width: 180px;
    height: fit-content;

    @media (min-width: ${breakpoints.MD}) {
      max-width: 100%;
    }
  `,
};
