// @algolia/ui-library/public/layouts/Hero/Hero.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const breakpoints = __importStar(require("../../constants/breakpoints"));
const cloudinary_1 = __importDefault(require("../../constants/cloudinary"));
/**
 * To calculate the padding bottom
 * Open your SVG, and do (height/width)*100 = padding-bottom
 */
exports.default = {
    button: react_1.css `
    min-width: 200px;
  `,
    background: react_1.css `
    background-size: 100% 100%;
  `,
    imageRight: react_1.css `
    left: 50%;
    top: 50%;
    transform: translateX(120px) translateY(-50%);
    @media (min-width: ${breakpoints.LG}) {
      transform: translateX(280px) translateY(-50%);
    }
  `,
    imageLeft: react_1.css `
    right: 50%;
    top: 50%;
    transform: translateX(-120px) translateY(-50%);
    @media (min-width: ${breakpoints.LG}) {
      transform: translateX(-280px) translateY(-50%);
    }
  `,
    video: react_1.css `
    padding-bottom: 66%;
    @media (min-width: ${breakpoints.XS}) {
      padding-bottom: 46%;
    }
    @media (min-width: ${breakpoints.MD}) {
      padding-bottom: 66%;
    }
  `,
    videoTransform: react_1.css `
    @media (min-width: ${breakpoints.MD}) {
      transform: translateY(-50%);
    }
  `,
    videoMaxHeight: react_1.css `
    max-height: 300px;
  `,
    playButton: react_1.css `
    transform: translate(-50%, -50%);
  `,
    athleticFields: react_1.css `
    padding-bottom: 41%;
    background-image: url('${cloudinary_1.default}/hero/athletic-fields.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    bigCubes: react_1.css `
    padding-bottom: 121.25%;
    background-image: url('${cloudinary_1.default}/hero/big-cubes.svg');
    bottom: calc(-45vw + 95px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-45vw + 135px);
    }
  `,
    bubbles: react_1.css `
    padding-bottom: 35.06%;
    background-image: url('${cloudinary_1.default}/hero/bubbles.svg');
    bottom: calc(-18vw + 230px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-18vw + 300px);
    }
  `,
    circles: react_1.css `
    padding-bottom: 67.74%;
    background-image: url('${cloudinary_1.default}/hero/circles.svg');
    bottom: calc(-17.7vw + 103px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-17.7vw + 147px);
    }
  `,
    cloudLines: react_1.css `
    max-width: 650px;
    padding-bottom: 28%;
    background-image: url('${cloudinary_1.default}/hero/cloud-lines.svg');
    background-size: contain;
    background-repeat: no-repeat;
    top: 50%;
    transform: translateY(-50%);
  `,
    cubes: react_1.css `
    padding-bottom: 36.75%;
    background-image: url('${cloudinary_1.default}/hero/cubes.svg');
    bottom: calc(-20vw + 140px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-20vw + 180px);
    }
  `,
    curves: react_1.css `
    padding-bottom: 55.27%;
    background-image: url('${cloudinary_1.default}/hero/curves.svg');
    bottom: calc(-22.5vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-22.5vw + 142px);
    }
  `,
    docsearch: react_1.css `
    padding-bottom: 16.92%;
    background-image: url('${cloudinary_1.default}/hero/docsearch.svg');
    bottom: calc(-22.5vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-22.5vw + 142px);
    }
  `,
    graph: react_1.css `
    padding-bottom: 55.27%;
    background-image: url('${cloudinary_1.default}/hero/graph.svg');
    bottom: calc(-17vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-17vw + 142px);
    }
  `,
    graphBlueLine: react_1.css `
    padding-bottom: 55.27%;
    background-image: url('${cloudinary_1.default}/hero/graph-blue-line.svg');
    bottom: calc(-17vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-17vw + 142px);
    }
  `,
    intersections: react_1.css `
    top: 50%;
    padding-bottom: 115.49%;
    background-image: url('${cloudinary_1.default}/hero/intersections.svg');
    transform: translateY(-50%);
  `,
    map: react_1.css `
    padding-bottom: 59.375%;
    background-image: url('${cloudinary_1.default}/hero/map.svg');
    bottom: calc(-37vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-37vw + 140px);
    }
  `,
    orangeSquare: react_1.css `
    width: 954px;
    height: 500px;
    margin: auto;
    bottom: 100px;
    background-image: url('${cloudinary_1.default}/hero/orange-square.svg');
    @media (min-width: ${breakpoints.LG}) {
      width: 1336px;
      height: 700px;
      bottom: 140px;
    }
  `,
    orbit: react_1.css `
    top: 50%;
    padding-bottom: 55.27%;
    background-image: url('${cloudinary_1.default}/hero/orbit.svg');
    transform: translateY(-50%);
  `,
    orbInside: react_1.css `
    top: 50%;
    padding-bottom: 48.74%;
    background-image: url('${cloudinary_1.default}/hero/orb-inside.svg');
    transform: translateY(-50%);
  `,
    rectangles: react_1.css `
    top: 50%;
    padding-bottom: 48.6%;
    background-image: url('${cloudinary_1.default}/hero/rectangles.svg');
    transform: translateY(-50%);
  `,
    rectangles2: react_1.css `
    padding-bottom: 62.87%;
    background-image: url('${cloudinary_1.default}/hero/rectangle2.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    rounds: react_1.css `
    top: 50%;
    padding-bottom: 38.5%;
    background-image: url('${cloudinary_1.default}/hero/rounds.svg');
    transform: translateY(-50%);
  `,
    spirals: react_1.css `
    padding-bottom: 50%;
    background-image: url('${cloudinary_1.default}/hero/spirals.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    spotlights: react_1.css `
    padding-bottom: 55.27%;
    background-image: url('${cloudinary_1.default}/hero/spotlights.svg');
    bottom: calc(-22.5vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-22.5vw + 142px);
    }
  `,
    triangles: react_1.css `
    padding-bottom: 56.25%;
    background-image: url('${cloudinary_1.default}/hero/triangles.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    waves: react_1.css `
    padding-bottom: 19.125%;
    background-image: url('${cloudinary_1.default}/hero/waves.svg');
    bottom: calc(-10vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-10vw + 160px);
    }
  `,
    whirl: react_1.css `
    padding-bottom: 56.25%;
    background-image: url('${cloudinary_1.default}/hero/whirl.svg');
    bottom: calc(-25vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-25vw + 140px);
    }
  `,
};
