// @algolia/ui-library/public/esm/components/RelatedResources/RelatedResources.js
/** @jsxRuntime classic /
/** @jsx jsx */
import { useState } from 'react';
import cx from 'classnames';
import { css, jsx } from '@emotion/react';
import { Card, CardsGrid, Button, Heading3, LightCta, Text, TextBlock, } from '../../index';
import { Play } from 'react-feather';
import styles from './RelatedResources.css';
import { isInternalLink } from '../../helpers';
const buttonClass = 'uil-app-none uil-bdr-max uil-bgc-grey-800 uil-bdw-0 uil-color-white uil-d-inline-flex uil-ai-center uil-jc-center uil-w-30 uil-h-30 lg:uil-h-40 lg:uil-w-40';
/** `RelatedResources` is a media exposition component. It can display multiple cards. Each one with
 * a title, a description and an image. Clicking on a card redirects to a URL.
 *
 * `RelatedResources` is meant to be source agnostic. Meaning that no matter the source of the data, it will
 * properly display them.
 *
 * `RelatedResources` is primarly made to be used with GraphQL returned data
 */
const RelatedResources = ({ cardsPerPage, cardsProps = { tag: 'a' }, data, descriptionLines = 3, titleLines = 2, externalRoutes = [
    'dashboard',
    'doc',
    'ebook',
    'event',
    'lp',
    'pdfs',
    'study',
    'support',
    'thank-you',
    'users',
    'webinar',
], seeMore, ...other }) => {
    const [isAtStart, setIsAtStart] = useState(true);
    const [isAtEnd, setIsAtEnd] = useState(cardsPerPage >= data.length);
    const [start, setStart] = useState(0);
    const [end, setEnd] = useState(cardsPerPage);
    const [itemArray, setItemArray] = useState(data.slice(start, end));
    const setLastPage = () => {
        setIsAtEnd(true);
        setStart(data.length - cardsPerPage);
        setEnd(data.length);
        setItemArray(data.slice(data.length - cardsPerPage, data.length));
    };
    const setFirstPage = () => {
        setIsAtStart(true);
        setStart(0);
        setEnd(cardsPerPage);
        setItemArray(data.slice(0, cardsPerPage));
    };
    const setNewPage = (tmpStart, tmpEnd) => {
        setStart(tmpStart);
        setEnd(tmpEnd);
        setItemArray(data.slice(tmpStart, tmpEnd));
    };
    const next = () => {
        if (end < data.length) {
            setIsAtStart(false);
            const tmpStart = start + cardsPerPage;
            const tmpEnd = end + cardsPerPage;
            if (tmpEnd >= data.length)
                setLastPage();
            else
                setNewPage(tmpStart, tmpEnd);
        }
    };
    const prev = () => {
        if (start > 0) {
            setIsAtEnd(false);
            const tmpStart = start - cardsPerPage;
            const tmpEnd = end - cardsPerPage;
            if (tmpStart <= 0)
                setFirstPage();
            else
                setNewPage(tmpStart, tmpEnd);
        }
    };
    const clampStyles = (lines) => [
        styles.clamp,
        css `
      -webkit-line-clamp: ${lines};
      line-clamp: ${lines};
    `,
    ];
    return (jsx("div", Object.assign({}, other),
        jsx("div", { className: "uil-d-flex uil-jc-center uil-mb-12 lg:uil-mb-20 lg:uil-jc-end" },
            jsx("button", { "aria-label": "Previous page", className: cx('uil-mr-8 lg:uil-mr-12', buttonClass, isAtStart ? 'uil-op-50p' : 'uil-cursor-pointer'), onClick: prev },
                jsx(Play, { "aria-hidden": true, className: "uil-w-14 uil-h-14 uil-pos-relative", fill: "currentColor", css: css `
              left: -1px;
              transform: rotate(180deg);
            ` })),
            jsx("button", { "aria-label": "Next page", className: cx(buttonClass, isAtEnd ? 'uil-op-50p' : 'uil-cursor-pointer'), onClick: next },
                jsx(Play, { "aria-hidden": true, className: "uil-w-14 uil-h-14 uil-pos-relative", css: css `
              left: 1px;
            `, fill: "currentColor" }))),
        jsx(CardsGrid, { columnsPerRow: itemArray.length }, itemArray.map(node => {
            const textCta = node.type === 'videos' ? 'Watch video' : 'Read more';
            return (jsx(Card, { key: node.title, image: node.image, imageProps: {
                    alt: `illustration for: '${node.title}'"`,
                    loading: 'lazy',
                }, elevation: true, tag: cardsProps.tag, href: isInternalLink(node.url, externalRoutes) ? null : node.url, to: isInternalLink(node.url, externalRoutes) ? node.url : null, className: "uil-td-none uil-d-block" },
                jsx("div", { className: "uil-d-flex uil-fxd-column uil-jc-between uil-h-100p" },
                    node.description ? (jsx(TextBlock, { title: node.title, titleProps: {
                            className: 'uil-ov-hidden uil-to-ellipsis',
                            css: clampStyles(titleLines),
                        } },
                        jsx(Text, { tag: "div" },
                            jsx("p", { css: clampStyles(descriptionLines), className: "uil-m-0 uil-ov-hidden uil-to-ellipsis" }, node.description)))) : (jsx(Heading3, { className: "uil-ov-hidden uil-to-ellipsis uil-mv-0", css: clampStyles(titleLines) }, node.title)),
                    jsx("div", { className: "uil-js-end" },
                        node.author && (jsx("div", { className: "uil-d-flex uil-ai-center uil-mt-16" },
                            jsx("img", { src: node.author.avatar_url, alt: node.author.name, className: "uil-w-50 uil-h-50 uil-bdr-max uil-mr-16", loading: "lazy" }),
                            jsx("div", { className: "uil-fxg-1" },
                                jsx(Text, { className: "uil-m-0 uil-lh-big" }, node.author.name),
                                jsx(Text, { small: true, className: "uil-m-0 uil-fw-bold" }, node.author.job_title)))),
                        jsx(LightCta, { tag: "p", className: "uil-mt-16 uil-mb-0", withTail: true }, textCta)))));
        })),
        seeMore && (jsx("div", { className: "uil-ta-center uil-pt-24 lg:uil-pt-32" },
            jsx(Button, { href: seeMore, background: "white", color: "grey" }, "See more")))));
};
export default RelatedResources;
