// @algolia/ui-library/public/esm/layouts/Section/Section.css.js
import { css } from '@emotion/react';
import * as colors from '../../constants/colors';
import * as breakpoints from '../../constants/breakpoints';
export default {
    gradient: css `
    background-image: linear-gradient(${colors.WHITE}, ${colors.GREY_100});
  `,
    dark: css `
    background-image: radial-gradient(
      circle at 50% 0,
      ${colors.GREY_700},
      ${colors.GREY_900}
    );
  `,
    gradientSide: css `
    background-image: linear-gradient(
      to right,
      ${colors.GREY_100},
      ${colors.WHITE},
      ${colors.GREY_100}
    );
  `,
    gradientCenter: css `
    background: linear-gradient(
      ${colors.WHITE},
      ${colors.GREY_100},
      ${colors.WHITE}
    );
  `,
    split: css `
    @media (min-width: ${breakpoints.MD}) {
      background: linear-gradient(
        to right,
        ${colors.GREY_900} 50%,
        ${colors.GREY_100} 50%
      );
    }
  `,
    reverseSplit: css `
    @media (min-width: ${breakpoints.MD}) {
      background: linear-gradient(
        to right,
        ${colors.GREY_100} 50%,
        ${colors.GREY_900} 50%
      );
    }
  `,
};
