// @algolia/ui-library/public/components/Footer/Banner/Banner.css.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = require("@emotion/react");
const colors = __importStar(require("../../../constants/colors"));
const breakpoints = __importStar(require("../../../constants/breakpoints"));
exports.default = {
    default: react_1.css `
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.CYAN_400} 0%,
      ${colors.NEBULA_500} 100%
    );
  `,
    search: react_1.css `
    padding-bottom: 10rem;
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.CYAN_400} 0%,
      ${colors.NEBULA_500} 100%
    );
  `,
    answers: react_1.css `
    padding-bottom: 10rem;
    background: radial-gradient(
      100% 100% at 100% 0%,
      #cb97ff 0%,
      ${colors.COLOR_ANSWER} 100%
    );
  `,
    recommend: react_1.css `
    padding-bottom: 10rem;
    background: radial-gradient(
      100% 100% at 100% 0%,
      ${colors.PINK_300} 0%,
      ${colors.COLOR_RECOMMEND} 100%
    );
  `,
    card: react_1.css `
    background: linear-gradient(
      180deg,
      ${colors.GREY_600} 0%,
      ${colors.GREY_700} 100%
    );
    box-shadow: inset 0px -2px 4px rgba(0, 0, 0, 0.08);
  `,
    cardContainer: react_1.css `
    margin-top: -10rem;

    @media (min-width: ${breakpoints.SM}) {
      margin-top: -11rem;
    }
  `,
};
