// @algolia/ui-library/public/esm/layouts/Hero/Hero.css.js
import { css } from '@emotion/react';
import * as breakpoints from '../../constants/breakpoints';
import CLOUDINARY from '../../constants/cloudinary';
/**
 * To calculate the padding bottom
 * Open your SVG, and do (height/width)*100 = padding-bottom
 */
export default {
    button: css `
    min-width: 200px;
  `,
    background: css `
    background-size: 100% 100%;
  `,
    imageRight: css `
    left: 50%;
    top: 50%;
    transform: translateX(120px) translateY(-50%);
    @media (min-width: ${breakpoints.LG}) {
      transform: translateX(280px) translateY(-50%);
    }
  `,
    imageLeft: css `
    right: 50%;
    top: 50%;
    transform: translateX(-120px) translateY(-50%);
    @media (min-width: ${breakpoints.LG}) {
      transform: translateX(-280px) translateY(-50%);
    }
  `,
    video: css `
    padding-bottom: 66%;
    @media (min-width: ${breakpoints.XS}) {
      padding-bottom: 46%;
    }
    @media (min-width: ${breakpoints.MD}) {
      padding-bottom: 66%;
    }
  `,
    videoTransform: css `
    @media (min-width: ${breakpoints.MD}) {
      transform: translateY(-50%);
    }
  `,
    videoMaxHeight: css `
    max-height: 300px;
  `,
    playButton: css `
    transform: translate(-50%, -50%);
  `,
    athleticFields: css `
    padding-bottom: 41%;
    background-image: url('${CLOUDINARY}/hero/athletic-fields.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    bigCubes: css `
    padding-bottom: 121.25%;
    background-image: url('${CLOUDINARY}/hero/big-cubes.svg');
    bottom: calc(-45vw + 95px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-45vw + 135px);
    }
  `,
    bubbles: css `
    padding-bottom: 35.06%;
    background-image: url('${CLOUDINARY}/hero/bubbles.svg');
    bottom: calc(-18vw + 230px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-18vw + 300px);
    }
  `,
    circles: css `
    padding-bottom: 67.74%;
    background-image: url('${CLOUDINARY}/hero/circles.svg');
    bottom: calc(-17.7vw + 103px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-17.7vw + 147px);
    }
  `,
    cloudLines: css `
    max-width: 650px;
    padding-bottom: 28%;
    background-image: url('${CLOUDINARY}/hero/cloud-lines.svg');
    background-size: contain;
    background-repeat: no-repeat;
    top: 50%;
    transform: translateY(-50%);
  `,
    cubes: css `
    padding-bottom: 36.75%;
    background-image: url('${CLOUDINARY}/hero/cubes.svg');
    bottom: calc(-20vw + 140px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-20vw + 180px);
    }
  `,
    curves: css `
    padding-bottom: 55.27%;
    background-image: url('${CLOUDINARY}/hero/curves.svg');
    bottom: calc(-22.5vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-22.5vw + 142px);
    }
  `,
    docsearch: css `
    padding-bottom: 16.92%;
    background-image: url('${CLOUDINARY}/hero/docsearch.svg');
    bottom: calc(-22.5vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-22.5vw + 142px);
    }
  `,
    graph: css `
    padding-bottom: 55.27%;
    background-image: url('${CLOUDINARY}/hero/graph.svg');
    bottom: calc(-17vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-17vw + 142px);
    }
  `,
    graphBlueLine: css `
    padding-bottom: 55.27%;
    background-image: url('${CLOUDINARY}/hero/graph-blue-line.svg');
    bottom: calc(-17vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-17vw + 142px);
    }
  `,
    intersections: css `
    top: 50%;
    padding-bottom: 115.49%;
    background-image: url('${CLOUDINARY}/hero/intersections.svg');
    transform: translateY(-50%);
  `,
    map: css `
    padding-bottom: 59.375%;
    background-image: url('${CLOUDINARY}/hero/map.svg');
    bottom: calc(-37vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-37vw + 140px);
    }
  `,
    orangeSquare: css `
    width: 954px;
    height: 500px;
    margin: auto;
    bottom: 100px;
    background-image: url('${CLOUDINARY}/hero/orange-square.svg');
    @media (min-width: ${breakpoints.LG}) {
      width: 1336px;
      height: 700px;
      bottom: 140px;
    }
  `,
    orbit: css `
    top: 50%;
    padding-bottom: 55.27%;
    background-image: url('${CLOUDINARY}/hero/orbit.svg');
    transform: translateY(-50%);
  `,
    orbInside: css `
    top: 50%;
    padding-bottom: 48.74%;
    background-image: url('${CLOUDINARY}/hero/orb-inside.svg');
    transform: translateY(-50%);
  `,
    rectangles: css `
    top: 50%;
    padding-bottom: 48.6%;
    background-image: url('${CLOUDINARY}/hero/rectangles.svg');
    transform: translateY(-50%);
  `,
    rectangles2: css `
    padding-bottom: 62.87%;
    background-image: url('${CLOUDINARY}/hero/rectangle2.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    rounds: css `
    top: 50%;
    padding-bottom: 38.5%;
    background-image: url('${CLOUDINARY}/hero/rounds.svg');
    transform: translateY(-50%);
  `,
    spirals: css `
    padding-bottom: 50%;
    background-image: url('${CLOUDINARY}/hero/spirals.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    spotlights: css `
    padding-bottom: 55.27%;
    background-image: url('${CLOUDINARY}/hero/spotlights.svg');
    bottom: calc(-22.5vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-22.5vw + 142px);
    }
  `,
    triangles: css `
    padding-bottom: 56.25%;
    background-image: url('${CLOUDINARY}/hero/triangles.svg');
    top: 50%;
    transform: translateY(-50%);
  `,
    waves: css `
    padding-bottom: 19.125%;
    background-image: url('${CLOUDINARY}/hero/waves.svg');
    bottom: calc(-10vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-10vw + 160px);
    }
  `,
    whirl: css `
    padding-bottom: 56.25%;
    background-image: url('${CLOUDINARY}/hero/whirl.svg');
    bottom: calc(-25vw + 100px);
    @media (min-width: ${breakpoints.LG}) {
      bottom: calc(-25vw + 140px);
    }
  `,
};
